// @wix/auto_sdk_blog_categories/build/es/meta.mjs
// src/blog-v3-category-categories.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(opts) {
  const domainToMappings = {
    "manage._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/bulk/category",
        destPath: "/v3/bulk/category"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/bulk/categories",
        destPath: "/v3/bulk/categories"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "social-blog._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    _: [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/blog/v3/categories",
        destPath: "/v3/categories"
      },
      {
        srcPath: "/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/categories",
        destPath: "/v3/categories"
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/category-service-api-proxy",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_categories";
function createCategory(payload) {
  function __createCategory({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.updatedDate" },
          { path: "category.coverImage.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "category.coverImage.focalPoint.x" },
          { path: "category.coverImage.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.CreateCategory",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        { protoPath: "/v3/categories", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.updatedDate" },
            { path: "category.coverImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "category.coverImage.focalPoint.x" },
            { path: "category.coverImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createCategory;
}
function updateCategory(payload) {
  function __updateCategory({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "category.updatedDate" },
          { path: "category.coverImage.urlExpirationDate" }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "category.coverImage.focalPoint.x" },
          { path: "category.coverImage.focalPoint.y" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.UpdateCategory",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        {
          protoPath: "/v3/categories/{category.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.updatedDate" },
            { path: "category.coverImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "category.coverImage.focalPoint.x" },
            { path: "category.coverImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateCategory;
}
function getCategory(payload) {
  function __getCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.GetCategory",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        { protoPath: "/v3/categories/{categoryId}", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.updatedDate" },
            { path: "category.coverImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "category.coverImage.focalPoint.x" },
            { path: "category.coverImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCategory;
}
function getCategoryBySlug(payload) {
  function __getCategoryBySlug({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.GetCategoryBySlug",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        { protoPath: "/v3/categories/slugs/{slug}", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "category.updatedDate" },
            { path: "category.coverImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "category.coverImage.focalPoint.x" },
            { path: "category.coverImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getCategoryBySlug;
}
function listCategories(payload) {
  function __listCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.ListCategories",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        { protoPath: "/v3/categories", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.updatedDate" },
            { path: "categories.coverImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "categories.coverImage.focalPoint.x" },
            { path: "categories.coverImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listCategories;
}
function queryCategories(payload) {
  function __queryCategories({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.QueryCategories",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        { protoPath: "/v3/categories/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "categories.updatedDate" },
            { path: "categories.coverImage.urlExpirationDate" }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "categories.coverImage.focalPoint.x" },
            { path: "categories.coverImage.focalPoint.y" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryCategories;
}
function deleteCategory(payload) {
  function __deleteCategory({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.category",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.CategoryService.DeleteCategory",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3CategoryServiceUrl(
        { protoPath: "/v3/categories/{categoryId}", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteCategory;
}

// src/blog-v3-category-categories.meta.ts
function createCategory2() {
  const payload = {};
  const getRequestOptions = createCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateCategory2() {
  const payload = { category: { id: ":categoryId" } };
  const getRequestOptions = updateCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/categories/{category.id}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = getCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getCategoryBySlug2() {
  const payload = { slug: ":slug" };
  const getRequestOptions = getCategoryBySlug(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/categories/slugs/{slug}",
    pathParams: { slug: "slug" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listCategories2() {
  const payload = {};
  const getRequestOptions = listCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/categories",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryCategories2() {
  const payload = {};
  const getRequestOptions = queryCategories(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/categories/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteCategory2() {
  const payload = { categoryId: ":categoryId" };
  const getRequestOptions = deleteCategory(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/categories/{categoryId}",
    pathParams: { categoryId: "categoryId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  createCategory2 as createCategory,
  deleteCategory2 as deleteCategory,
  getCategory2 as getCategory,
  getCategoryBySlug2 as getCategoryBySlug,
  listCategories2 as listCategories,
  queryCategories2 as queryCategories,
  updateCategory2 as updateCategory
};
