// @wix/auto_sdk_blog_draft-posts/build/internal/es/meta.mjs
// src/blog-v3-draft-draft-posts.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformSDKFloatToRESTFloat } from "@wix/sdk-runtime/transformations/float";
import { transformRESTFloatToSDKFloat } from "@wix/sdk-runtime/transformations/float";
import { transformSDKTimestampToRESTTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformSDKFieldMaskToRESTFieldMask } from "@wix/sdk-runtime/transformations/field-mask";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(opts) {
  const domainToMappings = {
    _api_base_domain_: [
      {
        srcPath: "/draft-post-service-api-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    _: [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/_api/draft-post-service-api-proxy/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "social-blog._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/blog/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts/create",
        destPath: "/v3/bulk/draft-posts/create"
      },
      {
        srcPath: "/blog/v3/draft-posts/",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_draft-posts";
function createDraftPost(payload) {
  function __createDraftPost({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "draftPost.editedDate" },
          { path: "draftPost.scheduledPublishDate" },
          { path: "draftPost.firstPublishedDate" },
          { path: "draftPost.createdDate" },
          { path: "draftPost.coverMedia.image.urlExpirationDate" },
          { path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate" },
          { path: "draftPost.heroImage.urlExpirationDate" },
          { path: "draftPost.richContent.metadata.createdTimestamp" },
          { path: "draftPost.richContent.metadata.updatedTimestamp" },
          { path: "draftPost.moderationDetails.submittedDate" },
          { path: "draftPost.moderationDetails.moderationDate" },
          { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPost.coverMedia.image.focalPoint.x" },
          { path: "draftPost.coverMedia.image.focalPoint.y" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
          { path: "draftPost.heroImage.focalPoint.x" },
          { path: "draftPost.heroImage.focalPoint.y" },
          { path: "draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPost.richContent.nodes.imageData.image.duration" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPost.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPost.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPost.richContent.nodes.videoData.video.duration" },
          { path: "draftPost.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPost.richContent.nodes.audioData.audio.duration" },
          { path: "draftPost.richContent.nodes.audioData.coverImage.duration" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.CreateDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDraftPost;
}
function bulkCreateDraftPosts(payload) {
  function __bulkCreateDraftPosts({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "draftPosts.editedDate" },
          { path: "draftPosts.scheduledPublishDate" },
          { path: "draftPosts.firstPublishedDate" },
          { path: "draftPosts.createdDate" },
          { path: "draftPosts.coverMedia.image.urlExpirationDate" },
          { path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate" },
          { path: "draftPosts.heroImage.urlExpirationDate" },
          { path: "draftPosts.richContent.metadata.createdTimestamp" },
          { path: "draftPosts.richContent.metadata.updatedTimestamp" },
          { path: "draftPosts.moderationDetails.submittedDate" },
          { path: "draftPosts.moderationDetails.moderationDate" },
          { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPosts.coverMedia.image.focalPoint.x" },
          { path: "draftPosts.coverMedia.image.focalPoint.y" },
          { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
          { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
          { path: "draftPosts.heroImage.focalPoint.x" },
          { path: "draftPosts.heroImage.focalPoint.y" },
          { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
          { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPosts.richContent.nodes.imageData.image.duration" },
          { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPosts.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPosts.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPosts.richContent.nodes.videoData.video.duration" },
          { path: "draftPosts.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
          {
            path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkCreateDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/bulk/draft-posts/create", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.coverMedia.image.urlExpirationDate" },
            {
              path: "results.item.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.coverMedia.image.focalPoint.x" },
            { path: "results.item.coverMedia.image.focalPoint.y" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateDraftPosts;
}
function bulkUpdateDraftPosts(payload) {
  function __bulkUpdateDraftPosts({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "scheduledPublishDate" },
          { path: "draftPosts.draftPost.editedDate" },
          { path: "draftPosts.draftPost.scheduledPublishDate" },
          { path: "draftPosts.draftPost.firstPublishedDate" },
          { path: "draftPosts.draftPost.createdDate" },
          { path: "draftPosts.draftPost.coverMedia.image.urlExpirationDate" },
          {
            path: "draftPosts.draftPost.coverMedia.video.thumbnail.urlExpirationDate"
          },
          { path: "draftPosts.draftPost.heroImage.urlExpirationDate" },
          {
            path: "draftPosts.draftPost.richContent.metadata.createdTimestamp"
          },
          {
            path: "draftPosts.draftPost.richContent.metadata.updatedTimestamp"
          },
          { path: "draftPosts.draftPost.moderationDetails.submittedDate" },
          { path: "draftPosts.draftPost.moderationDetails.moderationDate" },
          {
            path: "draftPosts.draftPost.media.wixMedia.image.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "draftPosts.fieldMask" }]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPosts.draftPost.coverMedia.image.focalPoint.x" },
          { path: "draftPosts.draftPost.coverMedia.image.focalPoint.y" },
          {
            path: "draftPosts.draftPost.coverMedia.video.thumbnail.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.coverMedia.video.thumbnail.focalPoint.y"
          },
          { path: "draftPosts.draftPost.heroImage.focalPoint.x" },
          { path: "draftPosts.draftPost.heroImage.focalPoint.y" },
          { path: "draftPosts.draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPosts.draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.y"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.imageData.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lat"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lng"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.appEmbedData.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.videoData.video.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.videoData.thumbnail.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.audioData.audio.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.audioData.coverImage.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkUpdateDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/update", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.coverMedia.image.urlExpirationDate" },
            {
              path: "results.item.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.coverMedia.image.focalPoint.x" },
            { path: "results.item.coverMedia.image.focalPoint.y" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateDraftPosts;
}
function listDeletedDraftPosts(payload) {
  function __listDeletedDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDeletedDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/trash-bin", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.coverMedia.image.focalPoint.x" },
            { path: "draftPosts.coverMedia.image.focalPoint.y" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedDraftPosts;
}
function getDraftPost(payload) {
  function __getDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/{draftPostId}", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDraftPost;
}
function updateDraftPost(payload) {
  function __updateDraftPost({ host }) {
    const serializedData = transformPaths(payload, [
      {
        transformFn: transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "scheduledPublishDate" },
          { path: "draftPost.editedDate" },
          { path: "draftPost.scheduledPublishDate" },
          { path: "draftPost.firstPublishedDate" },
          { path: "draftPost.createdDate" },
          { path: "draftPost.coverMedia.image.urlExpirationDate" },
          { path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate" },
          { path: "draftPost.heroImage.urlExpirationDate" },
          { path: "draftPost.richContent.metadata.createdTimestamp" },
          { path: "draftPost.richContent.metadata.updatedTimestamp" },
          { path: "draftPost.moderationDetails.submittedDate" },
          { path: "draftPost.moderationDetails.moderationDate" },
          { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPost.coverMedia.image.focalPoint.x" },
          { path: "draftPost.coverMedia.image.focalPoint.y" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
          { path: "draftPost.heroImage.focalPoint.x" },
          { path: "draftPost.heroImage.focalPoint.y" },
          { path: "draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPost.richContent.nodes.imageData.image.duration" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPost.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPost.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPost.richContent.nodes.videoData.video.duration" },
          { path: "draftPost.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPost.richContent.nodes.audioData.audio.duration" },
          { path: "draftPost.richContent.nodes.audioData.coverImage.duration" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.UpdateDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/{draftPost.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDraftPost;
}
function deleteDraftPost(payload) {
  function __deleteDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.DeleteDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/{draftPostId}", data: payload, host }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __deleteDraftPost;
}
function removeFromTrashBin(payload) {
  function __removeFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RemoveFromTrashBin",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload)
    };
    return metadata;
  }
  return __removeFromTrashBin;
}
function bulkDeleteDraftPosts(payload) {
  function __bulkDeleteDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkDeleteDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/bulk/draft-posts", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.coverMedia.image.urlExpirationDate" },
            {
              path: "results.item.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.coverMedia.image.focalPoint.x" },
            { path: "results.item.coverMedia.image.focalPoint.y" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteDraftPosts;
}
function listDraftPosts(payload) {
  function __listDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts", data: payload, host }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.coverMedia.image.focalPoint.x" },
            { path: "draftPosts.coverMedia.image.focalPoint.y" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDraftPosts;
}
function getDeletedDraftPost(payload) {
  function __getDeletedDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDeletedDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}",
          data: payload,
          host
        }
      ),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeletedDraftPost;
}
function restoreFromTrashBin(payload) {
  function __restoreFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RestoreFromTrashBin",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}/restore",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __restoreFromTrashBin;
}
function queryDraftPosts(payload) {
  function __queryDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.QueryDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.coverMedia.image.focalPoint.x" },
            { path: "draftPosts.coverMedia.image.focalPoint.y" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryDraftPosts;
}
function publishDraftPost(payload) {
  function __publishDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.PublishDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/{draftPostId}/publish",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __publishDraftPost;
}

// src/blog-v3-draft-draft-posts.meta.ts
function createDraftPost2() {
  const payload = {};
  const getRequestOptions = createDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkCreateDraftPosts2() {
  const payload = {};
  const getRequestOptions = bulkCreateDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/bulk/draft-posts/create",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkUpdateDraftPosts2() {
  const payload = {};
  const getRequestOptions = bulkUpdateDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/draft-posts/update",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDeletedDraftPosts2() {
  const payload = {};
  const getRequestOptions = listDeletedDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts/trash-bin",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = getDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function updateDraftPost2() {
  const payload = { draftPost: { id: ":draftPostId" } };
  const getRequestOptions = updateDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "PATCH",
    path: "/v3/draft-posts/{draftPost.id}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function deleteDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = deleteDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/draft-posts/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function removeFromTrashBin2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = removeFromTrashBin(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/draft-posts/trash-bin/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function bulkDeleteDraftPosts2() {
  const payload = {};
  const getRequestOptions = bulkDeleteDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "DELETE",
    path: "/v3/bulk/draft-posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listDraftPosts2() {
  const payload = {};
  const getRequestOptions = listDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getDeletedDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = getDeletedDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/draft-posts/trash-bin/{draftPostId}",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function restoreFromTrashBin2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = restoreFromTrashBin(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts/trash-bin/{draftPostId}/restore",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryDraftPosts2() {
  const payload = {};
  const getRequestOptions = queryDraftPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function publishDraftPost2() {
  const payload = { draftPostId: ":draftPostId" };
  const getRequestOptions = publishDraftPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/draft-posts/{draftPostId}/publish",
    pathParams: { draftPostId: "draftPostId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  bulkCreateDraftPosts2 as bulkCreateDraftPosts,
  bulkDeleteDraftPosts2 as bulkDeleteDraftPosts,
  bulkUpdateDraftPosts2 as bulkUpdateDraftPosts,
  createDraftPost2 as createDraftPost,
  deleteDraftPost2 as deleteDraftPost,
  getDeletedDraftPost2 as getDeletedDraftPost,
  getDraftPost2 as getDraftPost,
  listDeletedDraftPosts2 as listDeletedDraftPosts,
  listDraftPosts2 as listDraftPosts,
  publishDraftPost2 as publishDraftPost,
  queryDraftPosts2 as queryDraftPosts,
  removeFromTrashBin2 as removeFromTrashBin,
  restoreFromTrashBin2 as restoreFromTrashBin,
  updateDraftPost2 as updateDraftPost
};
