// @wix/blog_app-extensions/build/es/dashboard.mjs
var __defProp = Object.defineProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};

// src/extensions/dashboard-page-builders.ts
var dashboard_page_builders_exports = {};
__export(dashboard_page_builders_exports, {
  blogAnalytics: () => blogAnalytics,
  blogCategories: () => blogCategories,
  blogComments: () => blogComments,
  blogMonetization: () => blogMonetization,
  blogOverview: () => blogOverview,
  blogPosts: () => blogPosts,
  blogSettings: () => blogSettings,
  blogTags: () => blogTags,
  blogTemplates: () => blogTemplates,
  blogWriters: () => blogWriters
});
function blogAnalytics() {
  return { pageId: "3272eb63-13c2-4e22-bf9f-66964d1d39a3" };
}
function blogPosts() {
  return { pageId: "c7b86028-b1d2-42b0-be1d-4c95b374fc7d" };
}
function blogWriters() {
  return { pageId: "b8c96d90-95d6-57de-9a97-0d44c4dfaa96" };
}
function blogTags() {
  return { pageId: "06313e49-80f7-498e-99cd-312eab83f391" };
}
function blogComments() {
  return { pageId: "77f77e2b-56b6-4fe5-8011-ed1ebfd95c0b" };
}
function blogMonetization() {
  return { pageId: "681cbbb9-800c-414f-856a-ad4be4e1b296" };
}
function blogTemplates() {
  return { pageId: "374b3e95-6860-4018-906b-211aa786450d" };
}
function blogCategories() {
  return { pageId: "fd3e1c66-89b0-44ee-b824-1a28cc1a6dde" };
}
function blogOverview() {
  return { pageId: "7c8f91b9-0b65-47ee-92eb-37c1f9a4d410" };
}
function blogSettings() {
  return { pageId: "2718bafd-5769-487e-b162-4b4d747edf89" };
}

// src/extensions/dashboard-pages-plugin-props.ts
var dashboard_pages_plugin_props_exports = {};
__export(dashboard_pages_plugin_props_exports, {
  BlogCategories: () => fd3e1c66_89b0_44ee_b824_1a28cc1a6dde_exports,
  BlogOverview: () => c8f91b9_0b65_47ee_92eb_37c1f9a4d410_exports,
  BlogPosts: () => c7b86028_b1d2_42b0_be1d_4c95b374fc7d_exports,
  BlogTags: () => e49_80f7_498e_99cd_312eab83f391_exports,
  Posts: () => a7f877e0_69ac_459f_8b5b_eb7c5e25bad2_exports
});

// src/extensions/dashboard-page-plugins/c7b86028-b1d2-42b0-be1d-4c95b374fc7d.ts
var c7b86028_b1d2_42b0_be1d_4c95b374fc7d_exports = {};

// src/extensions/dashboard-page-plugins/a7f877e0-69ac-459f-8b5b-eb7c5e25bad2.ts
var a7f877e0_69ac_459f_8b5b_eb7c5e25bad2_exports = {};

// src/extensions/dashboard-page-plugins/06313e49-80f7-498e-99cd-312eab83f391.ts
var e49_80f7_498e_99cd_312eab83f391_exports = {};

// src/extensions/dashboard-page-plugins/fd3e1c66-89b0-44ee-b824-1a28cc1a6dde.ts
var fd3e1c66_89b0_44ee_b824_1a28cc1a6dde_exports = {};

// src/extensions/dashboard-page-plugins/7c8f91b9-0b65-47ee-92eb-37c1f9a4d410.ts
var c8f91b9_0b65_47ee_92eb_37c1f9a4d410_exports = {};
export {
  dashboard_page_builders_exports as pages,
  dashboard_pages_plugin_props_exports as plugins
};
