// @wix/auto_sdk_blog_draft-posts/build/internal/cjs/index.typings.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// index.typings.ts
var index_typings_exports = {};
__export(index_typings_exports, {
  Action: () => Action,
  Alignment: () => Alignment,
  AppType: () => AppType,
  AspectRatio: () => AspectRatio,
  BackgroundType: () => BackgroundType,
  ButtonDataType: () => ButtonDataType,
  CardStylesType: () => CardStylesType,
  Crop: () => Crop,
  DecorationType: () => DecorationType,
  Direction: () => Direction,
  DividerDataAlignment: () => DividerDataAlignment,
  Field: () => Field,
  FontType: () => FontType,
  GIFType: () => GIFType,
  GetDraftPostsSort: () => GetDraftPostsSort,
  InitialExpandedItems: () => InitialExpandedItems,
  Layout: () => Layout,
  LayoutType: () => LayoutType,
  LineStyle: () => LineStyle,
  MapType: () => MapType,
  ModerationStatusStatus: () => ModerationStatusStatus,
  NodeType: () => NodeType,
  NullValue: () => NullValue,
  Orientation: () => Orientation,
  Origin: () => Origin,
  Placement: () => Placement,
  PluginContainerDataAlignment: () => PluginContainerDataAlignment,
  PollLayoutDirection: () => PollLayoutDirection,
  PollLayoutType: () => PollLayoutType,
  Position: () => Position,
  Resizing: () => Resizing,
  SortOrder: () => SortOrder,
  Source: () => Source,
  Status: () => Status,
  StylesPosition: () => StylesPosition,
  Target: () => Target,
  TextAlignment: () => TextAlignment,
  ThumbnailsAlignment: () => ThumbnailsAlignment,
  TotalDraftPostsGroupingField: () => TotalDraftPostsGroupingField,
  Type: () => Type,
  VerticalAlignment: () => VerticalAlignment,
  ViewMode: () => ViewMode,
  ViewRole: () => ViewRole,
  VoteRole: () => VoteRole,
  WebhookIdentityType: () => WebhookIdentityType,
  Width: () => Width,
  WidthType: () => WidthType,
  bulkCreateDraftPosts: () => bulkCreateDraftPosts2,
  bulkDeleteDraftPosts: () => bulkDeleteDraftPosts2,
  bulkUpdateDraftPosts: () => bulkUpdateDraftPosts2,
  createDraftPost: () => createDraftPost2,
  deleteDraftPost: () => deleteDraftPost2,
  getDeletedDraftPost: () => getDeletedDraftPost2,
  getDraftPost: () => getDraftPost2,
  listDeletedDraftPosts: () => listDeletedDraftPosts2,
  listDraftPosts: () => listDraftPosts2,
  publishDraftPost: () => publishDraftPost2,
  queryDraftPosts: () => queryDraftPosts2,
  removeFromTrashBin: () => removeFromTrashBin2,
  restoreFromTrashBin: () => restoreFromTrashBin2,
  updateDraftPost: () => updateDraftPost2
});
module.exports = __toCommonJS(index_typings_exports);

// src/blog-v3-draft-draft-posts.universal.ts
var import_transform_error = require("@wix/sdk-runtime/transform-error");
var import_query_builder = require("@wix/sdk-runtime/query-builder");
var import_rename_all_nested_keys = require("@wix/sdk-runtime/rename-all-nested-keys");

// src/blog-v3-draft-draft-posts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_float2 = require("@wix/sdk-runtime/transformations/float");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_field_mask = require("@wix/sdk-runtime/transformations/field-mask");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(opts) {
  const domainToMappings = {
    _api_base_domain_: [
      {
        srcPath: "/draft-post-service-api-proxy",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    _: [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/_api/draft-post-service-api-proxy/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "social-blog._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/blog/v3/draft-posts",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts/create",
        destPath: "/v3/bulk/draft-posts/create"
      },
      {
        srcPath: "/blog/v3/draft-posts/",
        destPath: "/v3/draft-posts"
      },
      {
        srcPath: "/blog/v3/bulk/draft-posts",
        destPath: "/v3/bulk/draft-posts"
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api/v3/draft-posts",
        destPath: "/v3/draft-posts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_draft-posts";
function createDraftPost(payload) {
  function __createDraftPost({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "draftPost.editedDate" },
          { path: "draftPost.scheduledPublishDate" },
          { path: "draftPost.firstPublishedDate" },
          { path: "draftPost.createdDate" },
          { path: "draftPost.coverMedia.image.urlExpirationDate" },
          { path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate" },
          { path: "draftPost.heroImage.urlExpirationDate" },
          { path: "draftPost.richContent.metadata.createdTimestamp" },
          { path: "draftPost.richContent.metadata.updatedTimestamp" },
          { path: "draftPost.moderationDetails.submittedDate" },
          { path: "draftPost.moderationDetails.moderationDate" },
          { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPost.coverMedia.image.focalPoint.x" },
          { path: "draftPost.coverMedia.image.focalPoint.y" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
          { path: "draftPost.heroImage.focalPoint.x" },
          { path: "draftPost.heroImage.focalPoint.y" },
          { path: "draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPost.richContent.nodes.imageData.image.duration" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPost.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPost.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPost.richContent.nodes.videoData.video.duration" },
          { path: "draftPost.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPost.richContent.nodes.audioData.audio.duration" },
          { path: "draftPost.richContent.nodes.audioData.coverImage.duration" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.CreateDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __createDraftPost;
}
function bulkCreateDraftPosts(payload) {
  function __bulkCreateDraftPosts({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "draftPosts.editedDate" },
          { path: "draftPosts.scheduledPublishDate" },
          { path: "draftPosts.firstPublishedDate" },
          { path: "draftPosts.createdDate" },
          { path: "draftPosts.coverMedia.image.urlExpirationDate" },
          { path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate" },
          { path: "draftPosts.heroImage.urlExpirationDate" },
          { path: "draftPosts.richContent.metadata.createdTimestamp" },
          { path: "draftPosts.richContent.metadata.updatedTimestamp" },
          { path: "draftPosts.moderationDetails.submittedDate" },
          { path: "draftPosts.moderationDetails.moderationDate" },
          { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPosts.coverMedia.image.focalPoint.x" },
          { path: "draftPosts.coverMedia.image.focalPoint.y" },
          { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
          { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
          { path: "draftPosts.heroImage.focalPoint.x" },
          { path: "draftPosts.heroImage.focalPoint.y" },
          { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
          { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPosts.richContent.nodes.imageData.image.duration" },
          { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPosts.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPosts.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPosts.richContent.nodes.videoData.video.duration" },
          { path: "draftPosts.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
          {
            path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkCreateDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/bulk/draft-posts/create", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.coverMedia.image.urlExpirationDate" },
            {
              path: "results.item.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.coverMedia.image.focalPoint.x" },
            { path: "results.item.coverMedia.image.focalPoint.y" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkCreateDraftPosts;
}
function bulkUpdateDraftPosts(payload) {
  function __bulkUpdateDraftPosts({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "scheduledPublishDate" },
          { path: "draftPosts.draftPost.editedDate" },
          { path: "draftPosts.draftPost.scheduledPublishDate" },
          { path: "draftPosts.draftPost.firstPublishedDate" },
          { path: "draftPosts.draftPost.createdDate" },
          { path: "draftPosts.draftPost.coverMedia.image.urlExpirationDate" },
          {
            path: "draftPosts.draftPost.coverMedia.video.thumbnail.urlExpirationDate"
          },
          { path: "draftPosts.draftPost.heroImage.urlExpirationDate" },
          {
            path: "draftPosts.draftPost.richContent.metadata.createdTimestamp"
          },
          {
            path: "draftPosts.draftPost.richContent.metadata.updatedTimestamp"
          },
          { path: "draftPosts.draftPost.moderationDetails.submittedDate" },
          { path: "draftPosts.draftPost.moderationDetails.moderationDate" },
          {
            path: "draftPosts.draftPost.media.wixMedia.image.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "draftPosts.fieldMask" }]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPosts.draftPost.coverMedia.image.focalPoint.x" },
          { path: "draftPosts.draftPost.coverMedia.image.focalPoint.y" },
          {
            path: "draftPosts.draftPost.coverMedia.video.thumbnail.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.coverMedia.video.thumbnail.focalPoint.y"
          },
          { path: "draftPosts.draftPost.heroImage.focalPoint.x" },
          { path: "draftPosts.draftPost.heroImage.focalPoint.y" },
          { path: "draftPosts.draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPosts.draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.x"
          },
          {
            path: "draftPosts.draftPost.media.wixMedia.videoV2.posters.focalPoint.y"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.imageData.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lat"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.mapData.mapSettings.lng"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.poll.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.appEmbedData.image.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.videoData.video.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.videoData.thumbnail.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.audioData.audio.duration"
          },
          {
            path: "draftPosts.draftPost.richContent.nodes.audioData.coverImage.duration"
          }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkUpdateDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/update", data: serializedData, host }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.coverMedia.image.urlExpirationDate" },
            {
              path: "results.item.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.coverMedia.image.focalPoint.x" },
            { path: "results.item.coverMedia.image.focalPoint.y" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkUpdateDraftPosts;
}
function listDeletedDraftPosts(payload) {
  function __listDeletedDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDeletedDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/trash-bin", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.coverMedia.image.focalPoint.x" },
            { path: "draftPosts.coverMedia.image.focalPoint.y" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDeletedDraftPosts;
}
function getDraftPost(payload) {
  function __getDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/{draftPostId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDraftPost;
}
function updateDraftPost(payload) {
  function __updateDraftPost({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_field_mask.transformSDKFieldMaskToRESTFieldMask,
        paths: [{ path: "fieldMask" }]
      },
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [
          { path: "scheduledPublishDate" },
          { path: "draftPost.editedDate" },
          { path: "draftPost.scheduledPublishDate" },
          { path: "draftPost.firstPublishedDate" },
          { path: "draftPost.createdDate" },
          { path: "draftPost.coverMedia.image.urlExpirationDate" },
          { path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate" },
          { path: "draftPost.heroImage.urlExpirationDate" },
          { path: "draftPost.richContent.metadata.createdTimestamp" },
          { path: "draftPost.richContent.metadata.updatedTimestamp" },
          { path: "draftPost.moderationDetails.submittedDate" },
          { path: "draftPost.moderationDetails.moderationDate" },
          { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
          { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
          }
        ]
      },
      {
        transformFn: import_float.transformSDKFloatToRESTFloat,
        paths: [
          { path: "draftPost.coverMedia.image.focalPoint.x" },
          { path: "draftPost.coverMedia.image.focalPoint.y" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
          { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
          { path: "draftPost.heroImage.focalPoint.x" },
          { path: "draftPost.heroImage.focalPoint.y" },
          { path: "draftPost.media.wixMedia.image.focalPoint.x" },
          { path: "draftPost.media.wixMedia.image.focalPoint.y" },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
          },
          {
            path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
          },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
          { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
          {
            path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
          },
          {
            path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
          },
          { path: "draftPost.richContent.nodes.imageData.image.duration" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
          { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
          { path: "draftPost.richContent.nodes.pollData.poll.image.duration" },
          {
            path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
          },
          {
            path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
          },
          { path: "draftPost.richContent.nodes.appEmbedData.image.duration" },
          { path: "draftPost.richContent.nodes.videoData.video.duration" },
          { path: "draftPost.richContent.nodes.videoData.thumbnail.duration" },
          { path: "draftPost.richContent.nodes.audioData.audio.duration" },
          { path: "draftPost.richContent.nodes.audioData.coverImage.duration" }
        ]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "PATCH",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.UpdateDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/{draftPost.id}",
          data: serializedData,
          host
        }
      ),
      data: serializedData,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __updateDraftPost;
}
function deleteDraftPost(payload) {
  function __deleteDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.DeleteDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/{draftPostId}", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __deleteDraftPost;
}
function removeFromTrashBin(payload) {
  function __removeFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RemoveFromTrashBin",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __removeFromTrashBin;
}
function bulkDeleteDraftPosts(payload) {
  function __bulkDeleteDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "DELETE",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.BulkDeleteDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/bulk/draft-posts", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "results.item.editedDate" },
            { path: "results.item.scheduledPublishDate" },
            { path: "results.item.firstPublishedDate" },
            { path: "results.item.createdDate" },
            { path: "results.item.coverMedia.image.urlExpirationDate" },
            {
              path: "results.item.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "results.item.heroImage.urlExpirationDate" },
            { path: "results.item.richContent.metadata.createdTimestamp" },
            { path: "results.item.richContent.metadata.updatedTimestamp" },
            { path: "results.item.moderationDetails.submittedDate" },
            { path: "results.item.moderationDetails.moderationDate" },
            { path: "results.item.media.wixMedia.image.urlExpirationDate" },
            { path: "results.item.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "results.item.coverMedia.image.focalPoint.x" },
            { path: "results.item.coverMedia.image.focalPoint.y" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "results.item.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "results.item.heroImage.focalPoint.x" },
            { path: "results.item.heroImage.focalPoint.y" },
            { path: "results.item.media.wixMedia.image.focalPoint.x" },
            { path: "results.item.media.wixMedia.image.focalPoint.y" },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "results.item.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.galleryData.options.item.ratio"
            },
            {
              path: "results.item.richContent.nodes.imageData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lat"
            },
            {
              path: "results.item.richContent.nodes.mapData.mapSettings.lng"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "results.item.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "results.item.richContent.nodes.appEmbedData.image.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.video.duration"
            },
            {
              path: "results.item.richContent.nodes.videoData.thumbnail.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.audio.duration"
            },
            {
              path: "results.item.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __bulkDeleteDraftPosts;
}
function listDraftPosts(payload) {
  function __listDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.ListDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts", data: payload, host }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.coverMedia.image.focalPoint.x" },
            { path: "draftPosts.coverMedia.image.focalPoint.y" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listDraftPosts;
}
function getDeletedDraftPost(payload) {
  function __getDeletedDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.GetDeletedDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}",
          data: payload,
          host
        }
      ),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getDeletedDraftPost;
}
function restoreFromTrashBin(payload) {
  function __restoreFromTrashBin({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.RestoreFromTrashBin",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/trash-bin/{draftPostId}/restore",
          data: payload,
          host
        }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPost.editedDate" },
            { path: "draftPost.scheduledPublishDate" },
            { path: "draftPost.firstPublishedDate" },
            { path: "draftPost.createdDate" },
            { path: "draftPost.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPost.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPost.heroImage.urlExpirationDate" },
            { path: "draftPost.richContent.metadata.createdTimestamp" },
            { path: "draftPost.richContent.metadata.updatedTimestamp" },
            { path: "draftPost.moderationDetails.submittedDate" },
            { path: "draftPost.moderationDetails.moderationDate" },
            { path: "draftPost.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPost.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPost.coverMedia.image.focalPoint.x" },
            { path: "draftPost.coverMedia.image.focalPoint.y" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPost.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPost.heroImage.focalPoint.x" },
            { path: "draftPost.heroImage.focalPoint.y" },
            { path: "draftPost.media.wixMedia.image.focalPoint.x" },
            { path: "draftPost.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPost.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "draftPost.media.wixMedia.videoV2.posters.focalPoint.y" },
            {
              path: "draftPost.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPost.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPost.richContent.nodes.imageData.image.duration" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPost.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPost.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPost.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPost.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPost.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPost.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPost.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __restoreFromTrashBin;
}
function queryDraftPosts(payload) {
  function __queryDraftPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.QueryDraftPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        { protoPath: "/v3/draft-posts/query", data: payload, host }
      ),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "draftPosts.editedDate" },
            { path: "draftPosts.scheduledPublishDate" },
            { path: "draftPosts.firstPublishedDate" },
            { path: "draftPosts.createdDate" },
            { path: "draftPosts.coverMedia.image.urlExpirationDate" },
            {
              path: "draftPosts.coverMedia.video.thumbnail.urlExpirationDate"
            },
            { path: "draftPosts.heroImage.urlExpirationDate" },
            { path: "draftPosts.richContent.metadata.createdTimestamp" },
            { path: "draftPosts.richContent.metadata.updatedTimestamp" },
            { path: "draftPosts.moderationDetails.submittedDate" },
            { path: "draftPosts.moderationDetails.moderationDate" },
            { path: "draftPosts.media.wixMedia.image.urlExpirationDate" },
            { path: "draftPosts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_float2.transformRESTFloatToSDKFloat,
          paths: [
            { path: "draftPosts.coverMedia.image.focalPoint.x" },
            { path: "draftPosts.coverMedia.image.focalPoint.y" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "draftPosts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "draftPosts.heroImage.focalPoint.x" },
            { path: "draftPosts.heroImage.focalPoint.y" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.x" },
            { path: "draftPosts.media.wixMedia.image.focalPoint.y" },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.x"
            },
            {
              path: "draftPosts.media.wixMedia.videoV2.posters.focalPoint.y"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "draftPosts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "draftPosts.richContent.nodes.imageData.image.duration" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "draftPosts.richContent.nodes.mapData.mapSettings.lng" },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            {
              path: "draftPosts.richContent.nodes.appEmbedData.image.duration"
            },
            { path: "draftPosts.richContent.nodes.videoData.video.duration" },
            {
              path: "draftPosts.richContent.nodes.videoData.thumbnail.duration"
            },
            { path: "draftPosts.richContent.nodes.audioData.audio.duration" },
            {
              path: "draftPosts.richContent.nodes.audioData.coverImage.duration"
            }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryDraftPosts;
}
function publishDraftPost(payload) {
  function __publishDraftPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.draft",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.DraftPostService.PublishDraftPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3DraftPostServiceUrl(
        {
          protoPath: "/v3/draft-posts/{draftPostId}/publish",
          data: payload,
          host
        }
      ),
      data: payload
    };
    return metadata;
  }
  return __publishDraftPost;
}

// src/blog-v3-draft-draft-posts.universal.ts
var import_image = require("@wix/sdk-runtime/transformations/image");
var import_image2 = require("@wix/sdk-runtime/transformations/image");
var import_video_v2 = require("@wix/sdk-runtime/transformations/video-v2");
var import_video_v22 = require("@wix/sdk-runtime/transformations/video-v2");
var import_video = require("@wix/sdk-runtime/transformations/video");
var import_video2 = require("@wix/sdk-runtime/transformations/video");
var import_page_url = require("@wix/sdk-runtime/transformations/page-url");
var import_page_url2 = require("@wix/sdk-runtime/transformations/page-url");
var import_transform_paths2 = require("@wix/sdk-runtime/transformations/transform-paths");
var Origin = /* @__PURE__ */ ((Origin2) => {
  Origin2["UNKNOWN"] = "UNKNOWN";
  Origin2["ADMIN"] = "ADMIN";
  Origin2["ADD_CATEGORIES"] = "ADD_CATEGORIES";
  Origin2["AUTO_SAVE"] = "AUTO_SAVE";
  Origin2["COPY_TEMPLATE"] = "COPY_TEMPLATE";
  Origin2["IMPORT"] = "IMPORT";
  Origin2["IMPORT_BULK"] = "IMPORT_BULK";
  Origin2["IMPORT_HTML"] = "IMPORT_HTML";
  Origin2["IMPORT_PATCH"] = "IMPORT_PATCH";
  Origin2["LANGUAGE_CHANGE"] = "LANGUAGE_CHANGE";
  Origin2["MANUAL_SAVE"] = "MANUAL_SAVE";
  Origin2["MIGRATION"] = "MIGRATION";
  Origin2["MODERATION"] = "MODERATION";
  Origin2["MOVE_TO_TRASH"] = "MOVE_TO_TRASH";
  Origin2["PRICING_PLANS_CHANGE"] = "PRICING_PLANS_CHANGE";
  Origin2["PROVISION"] = "PROVISION";
  Origin2["PUBLISH"] = "PUBLISH";
  Origin2["REASSIGN_OWNER"] = "REASSIGN_OWNER";
  Origin2["REBLOG"] = "REBLOG";
  Origin2["RESTORE"] = "RESTORE";
  Origin2["REVERT_TO_DRAFT"] = "REVERT_TO_DRAFT";
  Origin2["TRANSLATION"] = "TRANSLATION";
  Origin2["UNPUBLISH"] = "UNPUBLISH";
  Origin2["UNSCHEDULE"] = "UNSCHEDULE";
  Origin2["NEW_EDIT_SESSION"] = "NEW_EDIT_SESSION";
  Origin2["SCHEDULING_SERVICE_SCHEDULE"] = "SCHEDULING_SERVICE_SCHEDULE";
  Origin2["SCHEDULING_SERVICE_UNSCHEDULE"] = "SCHEDULING_SERVICE_UNSCHEDULE";
  Origin2["SCHEDULING_SERVICE_PUBLISH"] = "SCHEDULING_SERVICE_PUBLISH";
  Origin2["SCHEDULE"] = "SCHEDULE";
  Origin2["REMOVE_FROM_MODERATION"] = "REMOVE_FROM_MODERATION";
  Origin2["REJECT_FROM_MODERATION"] = "REJECT_FROM_MODERATION";
  Origin2["APPROVE_IN_MODERATION"] = "APPROVE_IN_MODERATION";
  Origin2["DELETE_TAG"] = "DELETE_TAG";
  Origin2["PIN"] = "PIN";
  Origin2["UNPIN"] = "UNPIN";
  Origin2["AI_AUTO_SAVE"] = "AI_AUTO_SAVE";
  return Origin2;
})(Origin || {});
var NodeType = /* @__PURE__ */ ((NodeType2) => {
  NodeType2["PARAGRAPH"] = "PARAGRAPH";
  NodeType2["TEXT"] = "TEXT";
  NodeType2["HEADING"] = "HEADING";
  NodeType2["BULLETED_LIST"] = "BULLETED_LIST";
  NodeType2["ORDERED_LIST"] = "ORDERED_LIST";
  NodeType2["LIST_ITEM"] = "LIST_ITEM";
  NodeType2["BLOCKQUOTE"] = "BLOCKQUOTE";
  NodeType2["CODE_BLOCK"] = "CODE_BLOCK";
  NodeType2["VIDEO"] = "VIDEO";
  NodeType2["DIVIDER"] = "DIVIDER";
  NodeType2["FILE"] = "FILE";
  NodeType2["GALLERY"] = "GALLERY";
  NodeType2["GIF"] = "GIF";
  NodeType2["HTML"] = "HTML";
  NodeType2["IMAGE"] = "IMAGE";
  NodeType2["LINK_PREVIEW"] = "LINK_PREVIEW";
  NodeType2["MAP"] = "MAP";
  NodeType2["POLL"] = "POLL";
  NodeType2["APP_EMBED"] = "APP_EMBED";
  NodeType2["BUTTON"] = "BUTTON";
  NodeType2["COLLAPSIBLE_LIST"] = "COLLAPSIBLE_LIST";
  NodeType2["TABLE"] = "TABLE";
  NodeType2["EMBED"] = "EMBED";
  NodeType2["COLLAPSIBLE_ITEM"] = "COLLAPSIBLE_ITEM";
  NodeType2["COLLAPSIBLE_ITEM_TITLE"] = "COLLAPSIBLE_ITEM_TITLE";
  NodeType2["COLLAPSIBLE_ITEM_BODY"] = "COLLAPSIBLE_ITEM_BODY";
  NodeType2["TABLE_CELL"] = "TABLE_CELL";
  NodeType2["TABLE_ROW"] = "TABLE_ROW";
  NodeType2["EXTERNAL"] = "EXTERNAL";
  NodeType2["AUDIO"] = "AUDIO";
  NodeType2["CAPTION"] = "CAPTION";
  NodeType2["LAYOUT"] = "LAYOUT";
  NodeType2["LAYOUT_CELL"] = "LAYOUT_CELL";
  return NodeType2;
})(NodeType || {});
var WidthType = /* @__PURE__ */ ((WidthType2) => {
  WidthType2["CONTENT"] = "CONTENT";
  WidthType2["SMALL"] = "SMALL";
  WidthType2["ORIGINAL"] = "ORIGINAL";
  WidthType2["FULL_WIDTH"] = "FULL_WIDTH";
  return WidthType2;
})(WidthType || {});
var PluginContainerDataAlignment = /* @__PURE__ */ ((PluginContainerDataAlignment2) => {
  PluginContainerDataAlignment2["CENTER"] = "CENTER";
  PluginContainerDataAlignment2["LEFT"] = "LEFT";
  PluginContainerDataAlignment2["RIGHT"] = "RIGHT";
  return PluginContainerDataAlignment2;
})(PluginContainerDataAlignment || {});
var ButtonDataType = /* @__PURE__ */ ((ButtonDataType2) => {
  ButtonDataType2["LINK"] = "LINK";
  ButtonDataType2["ACTION"] = "ACTION";
  return ButtonDataType2;
})(ButtonDataType || {});
var Target = /* @__PURE__ */ ((Target2) => {
  Target2["SELF"] = "SELF";
  Target2["BLANK"] = "BLANK";
  Target2["PARENT"] = "PARENT";
  Target2["TOP"] = "TOP";
  return Target2;
})(Target || {});
var TextAlignment = /* @__PURE__ */ ((TextAlignment2) => {
  TextAlignment2["AUTO"] = "AUTO";
  TextAlignment2["LEFT"] = "LEFT";
  TextAlignment2["RIGHT"] = "RIGHT";
  TextAlignment2["CENTER"] = "CENTER";
  TextAlignment2["JUSTIFY"] = "JUSTIFY";
  return TextAlignment2;
})(TextAlignment || {});
var LineStyle = /* @__PURE__ */ ((LineStyle2) => {
  LineStyle2["SINGLE"] = "SINGLE";
  LineStyle2["DOUBLE"] = "DOUBLE";
  LineStyle2["DASHED"] = "DASHED";
  LineStyle2["DOTTED"] = "DOTTED";
  return LineStyle2;
})(LineStyle || {});
var Width = /* @__PURE__ */ ((Width2) => {
  Width2["LARGE"] = "LARGE";
  Width2["MEDIUM"] = "MEDIUM";
  Width2["SMALL"] = "SMALL";
  return Width2;
})(Width || {});
var DividerDataAlignment = /* @__PURE__ */ ((DividerDataAlignment2) => {
  DividerDataAlignment2["CENTER"] = "CENTER";
  DividerDataAlignment2["LEFT"] = "LEFT";
  DividerDataAlignment2["RIGHT"] = "RIGHT";
  return DividerDataAlignment2;
})(DividerDataAlignment || {});
var ViewMode = /* @__PURE__ */ ((ViewMode2) => {
  ViewMode2["NONE"] = "NONE";
  ViewMode2["FULL"] = "FULL";
  ViewMode2["MINI"] = "MINI";
  return ViewMode2;
})(ViewMode || {});
var LayoutType = /* @__PURE__ */ ((LayoutType2) => {
  LayoutType2["COLLAGE"] = "COLLAGE";
  LayoutType2["MASONRY"] = "MASONRY";
  LayoutType2["GRID"] = "GRID";
  LayoutType2["THUMBNAIL"] = "THUMBNAIL";
  LayoutType2["SLIDER"] = "SLIDER";
  LayoutType2["SLIDESHOW"] = "SLIDESHOW";
  LayoutType2["PANORAMA"] = "PANORAMA";
  LayoutType2["COLUMN"] = "COLUMN";
  LayoutType2["MAGIC"] = "MAGIC";
  LayoutType2["FULLSIZE"] = "FULLSIZE";
  return LayoutType2;
})(LayoutType || {});
var Orientation = /* @__PURE__ */ ((Orientation2) => {
  Orientation2["ROWS"] = "ROWS";
  Orientation2["COLUMNS"] = "COLUMNS";
  return Orientation2;
})(Orientation || {});
var Crop = /* @__PURE__ */ ((Crop2) => {
  Crop2["FILL"] = "FILL";
  Crop2["FIT"] = "FIT";
  return Crop2;
})(Crop || {});
var ThumbnailsAlignment = /* @__PURE__ */ ((ThumbnailsAlignment2) => {
  ThumbnailsAlignment2["TOP"] = "TOP";
  ThumbnailsAlignment2["RIGHT"] = "RIGHT";
  ThumbnailsAlignment2["BOTTOM"] = "BOTTOM";
  ThumbnailsAlignment2["LEFT"] = "LEFT";
  ThumbnailsAlignment2["NONE"] = "NONE";
  return ThumbnailsAlignment2;
})(ThumbnailsAlignment || {});
var GIFType = /* @__PURE__ */ ((GIFType2) => {
  GIFType2["NORMAL"] = "NORMAL";
  GIFType2["STICKER"] = "STICKER";
  return GIFType2;
})(GIFType || {});
var Source = /* @__PURE__ */ ((Source2) => {
  Source2["HTML"] = "HTML";
  Source2["ADSENSE"] = "ADSENSE";
  return Source2;
})(Source || {});
var StylesPosition = /* @__PURE__ */ ((StylesPosition2) => {
  StylesPosition2["START"] = "START";
  StylesPosition2["END"] = "END";
  StylesPosition2["TOP"] = "TOP";
  StylesPosition2["HIDDEN"] = "HIDDEN";
  return StylesPosition2;
})(StylesPosition || {});
var MapType = /* @__PURE__ */ ((MapType2) => {
  MapType2["ROADMAP"] = "ROADMAP";
  MapType2["SATELITE"] = "SATELITE";
  MapType2["HYBRID"] = "HYBRID";
  MapType2["TERRAIN"] = "TERRAIN";
  return MapType2;
})(MapType || {});
var ViewRole = /* @__PURE__ */ ((ViewRole2) => {
  ViewRole2["CREATOR"] = "CREATOR";
  ViewRole2["VOTERS"] = "VOTERS";
  ViewRole2["EVERYONE"] = "EVERYONE";
  return ViewRole2;
})(ViewRole || {});
var VoteRole = /* @__PURE__ */ ((VoteRole2) => {
  VoteRole2["SITE_MEMBERS"] = "SITE_MEMBERS";
  VoteRole2["ALL"] = "ALL";
  return VoteRole2;
})(VoteRole || {});
var PollLayoutType = /* @__PURE__ */ ((PollLayoutType2) => {
  PollLayoutType2["LIST"] = "LIST";
  PollLayoutType2["GRID"] = "GRID";
  return PollLayoutType2;
})(PollLayoutType || {});
var PollLayoutDirection = /* @__PURE__ */ ((PollLayoutDirection2) => {
  PollLayoutDirection2["LTR"] = "LTR";
  PollLayoutDirection2["RTL"] = "RTL";
  return PollLayoutDirection2;
})(PollLayoutDirection || {});
var BackgroundType = /* @__PURE__ */ ((BackgroundType2) => {
  BackgroundType2["COLOR"] = "COLOR";
  BackgroundType2["IMAGE"] = "IMAGE";
  BackgroundType2["GRADIENT"] = "GRADIENT";
  return BackgroundType2;
})(BackgroundType || {});
var DecorationType = /* @__PURE__ */ ((DecorationType2) => {
  DecorationType2["BOLD"] = "BOLD";
  DecorationType2["ITALIC"] = "ITALIC";
  DecorationType2["UNDERLINE"] = "UNDERLINE";
  DecorationType2["SPOILER"] = "SPOILER";
  DecorationType2["ANCHOR"] = "ANCHOR";
  DecorationType2["MENTION"] = "MENTION";
  DecorationType2["LINK"] = "LINK";
  DecorationType2["COLOR"] = "COLOR";
  DecorationType2["FONT_SIZE"] = "FONT_SIZE";
  DecorationType2["EXTERNAL"] = "EXTERNAL";
  DecorationType2["STRIKETHROUGH"] = "STRIKETHROUGH";
  DecorationType2["SUPERSCRIPT"] = "SUPERSCRIPT";
  DecorationType2["SUBSCRIPT"] = "SUBSCRIPT";
  return DecorationType2;
})(DecorationType || {});
var FontType = /* @__PURE__ */ ((FontType2) => {
  FontType2["PX"] = "PX";
  FontType2["EM"] = "EM";
  return FontType2;
})(FontType || {});
var Position = /* @__PURE__ */ ((Position2) => {
  Position2["START"] = "START";
  Position2["END"] = "END";
  Position2["TOP"] = "TOP";
  return Position2;
})(Position || {});
var AspectRatio = /* @__PURE__ */ ((AspectRatio2) => {
  AspectRatio2["SQUARE"] = "SQUARE";
  AspectRatio2["RECTANGLE"] = "RECTANGLE";
  return AspectRatio2;
})(AspectRatio || {});
var Resizing = /* @__PURE__ */ ((Resizing2) => {
  Resizing2["FILL"] = "FILL";
  Resizing2["FIT"] = "FIT";
  return Resizing2;
})(Resizing || {});
var Placement = /* @__PURE__ */ ((Placement2) => {
  Placement2["IMAGE"] = "IMAGE";
  Placement2["PRODUCT_INFO"] = "PRODUCT_INFO";
  return Placement2;
})(Placement || {});
var CardStylesType = /* @__PURE__ */ ((CardStylesType2) => {
  CardStylesType2["CONTAINED"] = "CONTAINED";
  CardStylesType2["FRAMELESS"] = "FRAMELESS";
  return CardStylesType2;
})(CardStylesType || {});
var Alignment = /* @__PURE__ */ ((Alignment2) => {
  Alignment2["START"] = "START";
  Alignment2["CENTER"] = "CENTER";
  Alignment2["END"] = "END";
  return Alignment2;
})(Alignment || {});
var Layout = /* @__PURE__ */ ((Layout2) => {
  Layout2["STACKED"] = "STACKED";
  Layout2["SIDE_BY_SIDE"] = "SIDE_BY_SIDE";
  return Layout2;
})(Layout || {});
var AppType = /* @__PURE__ */ ((AppType2) => {
  AppType2["PRODUCT"] = "PRODUCT";
  AppType2["EVENT"] = "EVENT";
  AppType2["BOOKING"] = "BOOKING";
  return AppType2;
})(AppType || {});
var InitialExpandedItems = /* @__PURE__ */ ((InitialExpandedItems2) => {
  InitialExpandedItems2["FIRST"] = "FIRST";
  InitialExpandedItems2["ALL"] = "ALL";
  InitialExpandedItems2["NONE"] = "NONE";
  return InitialExpandedItems2;
})(InitialExpandedItems || {});
var Direction = /* @__PURE__ */ ((Direction2) => {
  Direction2["LTR"] = "LTR";
  Direction2["RTL"] = "RTL";
  return Direction2;
})(Direction || {});
var VerticalAlignment = /* @__PURE__ */ ((VerticalAlignment2) => {
  VerticalAlignment2["TOP"] = "TOP";
  VerticalAlignment2["MIDDLE"] = "MIDDLE";
  VerticalAlignment2["BOTTOM"] = "BOTTOM";
  return VerticalAlignment2;
})(VerticalAlignment || {});
var NullValue = /* @__PURE__ */ ((NullValue2) => {
  NullValue2["NULL_VALUE"] = "NULL_VALUE";
  return NullValue2;
})(NullValue || {});
var Status = /* @__PURE__ */ ((Status2) => {
  Status2["UNKNOWN"] = "UNKNOWN";
  Status2["PUBLISHED"] = "PUBLISHED";
  Status2["UNPUBLISHED"] = "UNPUBLISHED";
  Status2["SCHEDULED"] = "SCHEDULED";
  Status2["DELETED"] = "DELETED";
  Status2["IN_REVIEW"] = "IN_REVIEW";
  return Status2;
})(Status || {});
var ModerationStatusStatus = /* @__PURE__ */ ((ModerationStatusStatus2) => {
  ModerationStatusStatus2["UNKNOWN"] = "UNKNOWN";
  ModerationStatusStatus2["APPROVED"] = "APPROVED";
  ModerationStatusStatus2["REJECTED"] = "REJECTED";
  return ModerationStatusStatus2;
})(ModerationStatusStatus || {});
var TotalDraftPostsGroupingField = /* @__PURE__ */ ((TotalDraftPostsGroupingField2) => {
  TotalDraftPostsGroupingField2["STATUS"] = "STATUS";
  TotalDraftPostsGroupingField2["LANGUAGE"] = "LANGUAGE";
  return TotalDraftPostsGroupingField2;
})(TotalDraftPostsGroupingField || {});
var WebhookIdentityType = /* @__PURE__ */ ((WebhookIdentityType2) => {
  WebhookIdentityType2["UNKNOWN"] = "UNKNOWN";
  WebhookIdentityType2["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
  WebhookIdentityType2["MEMBER"] = "MEMBER";
  WebhookIdentityType2["WIX_USER"] = "WIX_USER";
  WebhookIdentityType2["APP"] = "APP";
  return WebhookIdentityType2;
})(WebhookIdentityType || {});
var Type = /* @__PURE__ */ ((Type2) => {
  Type2["UNKNOWN"] = "UNKNOWN";
  Type2["MANUAL"] = "MANUAL";
  Type2["AUTO_SAVE"] = "AUTO_SAVE";
  Type2["PROVISIONING"] = "PROVISIONING";
  Type2["IMPORT"] = "IMPORT";
  Type2["AI_AUTO_SAVE"] = "AI_AUTO_SAVE";
  return Type2;
})(Type || {});
var Field = /* @__PURE__ */ ((Field2) => {
  Field2["UNKNOWN"] = "UNKNOWN";
  Field2["URL"] = "URL";
  Field2["CONTENT"] = "CONTENT";
  Field2["RICH_CONTENT"] = "RICH_CONTENT";
  Field2["GENERATED_EXCERPT"] = "GENERATED_EXCERPT";
  return Field2;
})(Field || {});
var Action = /* @__PURE__ */ ((Action2) => {
  Action2["UPDATE"] = "UPDATE";
  Action2["UPDATE_PUBLISH"] = "UPDATE_PUBLISH";
  Action2["UPDATE_SCHEDULE"] = "UPDATE_SCHEDULE";
  Action2["UPDATE_REVERT_TO_DRAFT"] = "UPDATE_REVERT_TO_DRAFT";
  Action2["UPDATE_CANCEL_SCHEDULE"] = "UPDATE_CANCEL_SCHEDULE";
  Action2["UPDATE_REJECT"] = "UPDATE_REJECT";
  Action2["UPDATE_PUBLICATION"] = "UPDATE_PUBLICATION";
  return Action2;
})(Action || {});
var GetDraftPostsSort = /* @__PURE__ */ ((GetDraftPostsSort2) => {
  GetDraftPostsSort2["EDITING_DATE_DESC"] = "EDITING_DATE_DESC";
  GetDraftPostsSort2["EDITING_DATE_ASC"] = "EDITING_DATE_ASC";
  return GetDraftPostsSort2;
})(GetDraftPostsSort || {});
var SortOrder = /* @__PURE__ */ ((SortOrder2) => {
  SortOrder2["ASC"] = "ASC";
  SortOrder2["DESC"] = "DESC";
  return SortOrder2;
})(SortOrder || {});
async function createDraftPost2(draftPost, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      {
        draftPost,
        publish: options?.publish,
        fieldsets: options?.fieldsets
      },
      ["draftPost.richContent"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "draftPost.heroImage" },
          { path: "draftPost.coverMedia.image" },
          { path: "draftPost.media.wixMedia.image" }
        ]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "draftPost.url" },
          { path: "draftPost.translations.url" }
        ]
      },
      {
        transformFn: import_video.transformSDKVideoToRESTVideo,
        paths: [{ path: "draftPost.coverMedia.video" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
      }
    ]
  );
  const reqOpts = createDraftPost(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPost.heroImage" },
            { path: "draftPost.coverMedia.image" },
            { path: "draftPost.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPost.url" },
            { path: "draftPost.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPost.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPost.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          draftPost: "$[0]",
          publish: "$[1].publish",
          fieldsets: "$[1].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["draftPost", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkCreateDraftPosts2(draftPosts, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      {
        draftPosts,
        publish: options?.publish,
        returnFullEntity: options?.returnFullEntity,
        fieldsets: options?.fieldsets
      },
      ["draftPosts.richContent"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "draftPosts.heroImage" },
          { path: "draftPosts.coverMedia.image" },
          { path: "draftPosts.media.wixMedia.image" }
        ]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "draftPosts.url" },
          { path: "draftPosts.translations.url" }
        ]
      },
      {
        transformFn: import_video.transformSDKVideoToRESTVideo,
        paths: [{ path: "draftPosts.coverMedia.video" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "draftPosts.media.wixMedia.videoV2" }]
      }
    ]
  );
  const reqOpts = bulkCreateDraftPosts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.heroImage" },
            { path: "results.item.coverMedia.image" },
            { path: "results.item.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "results.item.url" },
            { path: "results.item.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "results.item.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.media.wixMedia.videoV2" }]
        }
      ]),
      ["results.item.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          draftPosts: "$[0]",
          publish: "$[1].publish",
          returnFullEntity: "$[1].returnFullEntity",
          fieldsets: "$[1].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["draftPosts", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkUpdateDraftPosts2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      {
        draftPosts: options?.draftPosts,
        action: options?.action,
        scheduledPublishDate: options?.scheduledPublishDate,
        returnFullEntity: options?.returnFullEntity,
        fieldsets: options?.fieldsets
      },
      ["draftPosts.draftPost.richContent"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "draftPosts.draftPost.heroImage" },
          { path: "draftPosts.draftPost.coverMedia.image" },
          { path: "draftPosts.draftPost.media.wixMedia.image" }
        ]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "draftPosts.draftPost.url" },
          { path: "draftPosts.draftPost.translations.url" }
        ]
      },
      {
        transformFn: import_video.transformSDKVideoToRESTVideo,
        paths: [{ path: "draftPosts.draftPost.coverMedia.video" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "draftPosts.draftPost.media.wixMedia.videoV2" }]
      }
    ]
  );
  const reqOpts = bulkUpdateDraftPosts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.heroImage" },
            { path: "results.item.coverMedia.image" },
            { path: "results.item.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "results.item.url" },
            { path: "results.item.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "results.item.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.media.wixMedia.videoV2" }]
        }
      ]),
      ["results.item.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          draftPosts: "$[0].draftPosts",
          action: "$[0].action",
          scheduledPublishDate: "$[0].scheduledPublishDate",
          returnFullEntity: "$[0].returnFullEntity",
          fieldsets: "$[0].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDeletedDraftPosts2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    {
      language: options?.language,
      sort: options?.sort,
      paging: options?.paging,
      fieldsets: options?.fieldsets,
      draftPostIds: options?.draftPostIds
    },
    []
  );
  const reqOpts = listDeletedDraftPosts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPosts.heroImage" },
            { path: "draftPosts.coverMedia.image" },
            { path: "draftPosts.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPosts.url" },
            { path: "draftPosts.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPosts.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPosts.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPosts.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          language: "$[0].language",
          sort: "$[0].sort",
          paging: "$[0].paging",
          fieldsets: "$[0].fieldsets",
          draftPostIds: "$[0].draftPostIds"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getDraftPost2(draftPostId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { draftPostId, fieldsets: options?.fieldsets },
    []
  );
  const reqOpts = getDraftPost(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPost.heroImage" },
            { path: "draftPost.coverMedia.image" },
            { path: "draftPost.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPost.url" },
            { path: "draftPost.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPost.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPost.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          draftPostId: "$[0]",
          fieldsets: "$[1].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["draftPostId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function updateDraftPost2(_id, draftPost, options) {
  const { httpClient, sideEffects } = arguments[3];
  const payload = (0, import_transform_paths2.transformPaths)(
    (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
      {
        draftPost: { ...draftPost, id: _id },
        action: options?.action,
        scheduledPublishDate: options?.scheduledPublishDate,
        fieldsets: options?.fieldsets
      },
      ["draftPost.richContent"]
    ),
    [
      {
        transformFn: import_image.transformSDKImageToRESTImage,
        paths: [
          { path: "draftPost.heroImage" },
          { path: "draftPost.coverMedia.image" },
          { path: "draftPost.media.wixMedia.image" }
        ]
      },
      {
        transformFn: import_page_url.transformSDKPageURLToRESTPageURL,
        paths: [
          { path: "draftPost.url" },
          { path: "draftPost.translations.url" }
        ]
      },
      {
        transformFn: import_video.transformSDKVideoToRESTVideo,
        paths: [{ path: "draftPost.coverMedia.video" }]
      },
      {
        transformFn: import_video_v2.transformSDKVideoV2ToRESTVideoV2,
        paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
      }
    ]
  );
  const reqOpts = updateDraftPost(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPost.heroImage" },
            { path: "draftPost.coverMedia.image" },
            { path: "draftPost.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPost.url" },
            { path: "draftPost.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPost.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPost.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: { draftPost: "$[1]" },
        explicitPathsToArguments: {
          "draftPost.id": "$[0]",
          action: "$[2].action",
          scheduledPublishDate: "$[2].scheduledPublishDate",
          fieldsets: "$[2].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["_id", "draftPost", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function deleteDraftPost2(draftPostId, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { draftPostId, permanent: options?.permanent },
    []
  );
  const reqOpts = deleteDraftPost(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          draftPostId: "$[0]",
          permanent: "$[1].permanent"
        },
        singleArgumentUnchanged: false
      },
      ["draftPostId", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function removeFromTrashBin2(draftPostId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { draftPostId },
    []
  );
  const reqOpts = removeFromTrashBin(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { draftPostId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["draftPostId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function bulkDeleteDraftPosts2(postIds, options) {
  const { httpClient, sideEffects } = arguments[2];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { postIds, permanent: options?.permanent },
    []
  );
  const reqOpts = bulkDeleteDraftPosts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "results.item.heroImage" },
            { path: "results.item.coverMedia.image" },
            { path: "results.item.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "results.item.url" },
            { path: "results.item.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "results.item.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "results.item.media.wixMedia.videoV2" }]
        }
      ]),
      ["results.item.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          postIds: "$[0]",
          permanent: "$[1].permanent"
        },
        singleArgumentUnchanged: false
      },
      ["postIds", "options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function listDraftPosts2(options) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    {
      status: options?.status,
      language: options?.language,
      sort: options?.sort,
      paging: options?.paging,
      fieldsets: options?.fieldsets
    },
    []
  );
  const reqOpts = listDraftPosts(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPosts.heroImage" },
            { path: "draftPosts.coverMedia.image" },
            { path: "draftPosts.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPosts.url" },
            { path: "draftPosts.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPosts.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPosts.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPosts.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {
          status: "$[0].status",
          language: "$[0].language",
          sort: "$[0].sort",
          paging: "$[0].paging",
          fieldsets: "$[0].fieldsets"
        },
        singleArgumentUnchanged: false
      },
      ["options"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function getDeletedDraftPost2(draftPostId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { draftPostId },
    []
  );
  const reqOpts = getDeletedDraftPost(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPost.heroImage" },
            { path: "draftPost.coverMedia.image" },
            { path: "draftPost.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPost.url" },
            { path: "draftPost.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPost.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPost.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { draftPostId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["draftPostId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
async function restoreFromTrashBin2(draftPostId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { draftPostId },
    []
  );
  const reqOpts = restoreFromTrashBin(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
      (0, import_transform_paths2.transformPaths)(result.data, [
        {
          transformFn: import_image2.transformRESTImageToSDKImage,
          paths: [
            { path: "draftPost.heroImage" },
            { path: "draftPost.coverMedia.image" },
            { path: "draftPost.media.wixMedia.image" }
          ]
        },
        {
          transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
          paths: [
            { path: "draftPost.url" },
            { path: "draftPost.translations.url" }
          ]
        },
        {
          transformFn: import_video2.transformRESTVideoToSDKVideo,
          paths: [{ path: "draftPost.coverMedia.video" }]
        },
        {
          transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
          paths: [{ path: "draftPost.media.wixMedia.videoV2" }]
        }
      ]),
      ["draftPost.richContent"]
    );
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { draftPostId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["draftPostId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
function queryDraftPosts2(options) {
  const { httpClient, sideEffects } = arguments[1];
  return (0, import_query_builder.queryBuilder)({
    func: async (payload) => {
      const reqOpts = queryDraftPosts({
        ...payload,
        ...options ?? {}
      });
      sideEffects?.onSiteCall?.();
      try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return result;
      } catch (err) {
        sideEffects?.onError?.(err);
        throw err;
      }
    },
    requestTransformer: (query) => {
      const args = [query, options];
      return (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
        { ...args?.[1], query: args?.[0] },
        []
      );
    },
    responseTransformer: ({ data }) => {
      const transformedData = (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(
        (0, import_transform_paths2.transformPaths)(data, [
          {
            transformFn: import_image2.transformRESTImageToSDKImage,
            paths: [
              { path: "draftPosts.heroImage" },
              { path: "draftPosts.coverMedia.image" },
              { path: "draftPosts.media.wixMedia.image" }
            ]
          },
          {
            transformFn: import_page_url2.transformRESTPageURLToSDKPageURL,
            paths: [
              { path: "draftPosts.url" },
              { path: "draftPosts.translations.url" }
            ]
          },
          {
            transformFn: import_video2.transformRESTVideoToSDKVideo,
            paths: [{ path: "draftPosts.coverMedia.video" }]
          },
          {
            transformFn: import_video_v22.transformRESTVideoV2ToSDKVideoV2,
            paths: [{ path: "draftPosts.media.wixMedia.videoV2" }]
          }
        ]),
        ["draftPosts.richContent"]
      );
      return {
        items: transformedData?.draftPosts,
        pagingMetadata: transformedData?.pagingMetadata
      };
    },
    errorTransformer: (err) => {
      const transformedError = (0, import_transform_error.transformError)(err, {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { query: "$[0]" },
        singleArgumentUnchanged: false
      });
      throw transformedError;
    },
    pagingMethod: "CURSOR",
    transformationPaths: {}
  });
}
async function publishDraftPost2(draftPostId) {
  const { httpClient, sideEffects } = arguments[1];
  const payload = (0, import_rename_all_nested_keys.renameKeysFromSDKRequestToRESTRequest)(
    { draftPostId },
    []
  );
  const reqOpts = publishDraftPost(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return (0, import_rename_all_nested_keys.renameKeysFromRESTResponseToSDKResponse)(result.data, []);
  } catch (err) {
    const transformedError = (0, import_transform_error.transformError)(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: { draftPostId: "$[0]" },
        singleArgumentUnchanged: false
      },
      ["draftPostId"]
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Action,
  Alignment,
  AppType,
  AspectRatio,
  BackgroundType,
  ButtonDataType,
  CardStylesType,
  Crop,
  DecorationType,
  Direction,
  DividerDataAlignment,
  Field,
  FontType,
  GIFType,
  GetDraftPostsSort,
  InitialExpandedItems,
  Layout,
  LayoutType,
  LineStyle,
  MapType,
  ModerationStatusStatus,
  NodeType,
  NullValue,
  Orientation,
  Origin,
  Placement,
  PluginContainerDataAlignment,
  PollLayoutDirection,
  PollLayoutType,
  Position,
  Resizing,
  SortOrder,
  Source,
  Status,
  StylesPosition,
  Target,
  TextAlignment,
  ThumbnailsAlignment,
  TotalDraftPostsGroupingField,
  Type,
  VerticalAlignment,
  ViewMode,
  ViewRole,
  VoteRole,
  WebhookIdentityType,
  Width,
  WidthType,
  bulkCreateDraftPosts,
  bulkDeleteDraftPosts,
  bulkUpdateDraftPosts,
  createDraftPost,
  deleteDraftPost,
  getDeletedDraftPost,
  getDraftPost,
  listDeletedDraftPosts,
  listDraftPosts,
  publishDraftPost,
  queryDraftPosts,
  removeFromTrashBin,
  restoreFromTrashBin,
  updateDraftPost
});
