// @wix/blog/build/cjs/meta.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  blogCache: () => blogCache,
  blogImporter: () => blogImporter,
  categories: () => categories,
  draftPosts: () => draftPosts,
  posts: () => posts,
  tags: () => tags
});
module.exports = __toCommonJS(meta_exports);
var blogCache = __toESM(require("@wix/auto_sdk_blog_blog-cache/meta"));
var blogImporter = __toESM(require("@wix/auto_sdk_blog_blog-importer/meta"));
var categories = __toESM(require("@wix/auto_sdk_blog_categories/meta"));
var draftPosts = __toESM(require("@wix/auto_sdk_blog_draft-posts/meta"));
var posts = __toESM(require("@wix/auto_sdk_blog_posts/meta"));
var tags = __toESM(require("@wix/auto_sdk_blog_tags/meta"));
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  blogCache,
  blogImporter,
  categories,
  draftPosts,
  posts,
  tags
});
