// @wix/auto_sdk_blog_posts/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getPost: () => getPost2,
  getPostBySlug: () => getPostBySlug2,
  getPostMetrics: () => getPostMetrics2,
  getTotalPosts: () => getTotalPosts2,
  listPosts: () => listPosts2,
  queryPostCountStats: () => queryPostCountStats2,
  queryPosts: () => queryPosts2
});
module.exports = __toCommonJS(meta_exports);

// src/blog-v3-post-posts.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_float = require("@wix/sdk-runtime/transformations/float");
var import_bytes = require("@wix/sdk-runtime/transformations/bytes");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_timestamp2 = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveComWixpressNpmCommunitiesPlatformizedBlogV3PostServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/communities-blog-node-api",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v2/stats",
        destPath: "/v2/stats"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v2/stats",
        destPath: "/v2/stats"
      }
    ],
    _: [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/blogs/this",
        destPath: "/_api/blogs/this"
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/categories",
        destPath: "/_api/categories"
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/site-properties",
        destPath: "/_api/site-properties"
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/posts/content",
        destPath: "/_api/posts/content"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v2/stats",
        destPath: "/v2/stats"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "social-blog._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/blog/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/blog/v3",
        destPath: "/v3"
      },
      {
        srcPath: "/v2/stats",
        destPath: "/v2/stats"
      },
      {
        srcPath: "/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/blog/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/blog/v2/stats",
        destPath: "/v2/stats"
      },
      {
        srcPath: "/_api/v3/posts",
        destPath: "/v3/posts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
function resolveComWixpressNpmCommunitiesPlatformizedBlogBlogStatsServiceUrl(opts) {
  const domainToMappings = {
    "api._api_base_domain_": [
      {
        srcPath: "/communities-blog-node-api",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v2/stats",
        destPath: "/v2/stats"
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v2/stats",
        destPath: "/v2/stats"
      }
    ],
    _: [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/blogs/this",
        destPath: "/_api/blogs/this"
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/categories",
        destPath: "/_api/categories"
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/site-properties",
        destPath: "/_api/site-properties"
      },
      {
        srcPath: "/_api/communities-blog-node-api/_api/posts/content",
        destPath: "/_api/posts/content"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/_api/communities-blog-node-api/v2/stats",
        destPath: "/v2/stats"
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "*.pub.wix-code.com": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "bo._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      },
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "social-blog._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "editor.wixapps.net": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "dev._base_domain_": [
      {
        srcPath: "/_api/communities-blog-node-api",
        destPath: ""
      }
    ],
    "www.wixapis.com": [
      {
        srcPath: "/blog/v2",
        destPath: "/v2"
      },
      {
        srcPath: "/blog/v3",
        destPath: "/v3"
      },
      {
        srcPath: "/v2/stats",
        destPath: "/v2/stats"
      },
      {
        srcPath: "/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/blog/v3/posts",
        destPath: "/v3/posts"
      },
      {
        srcPath: "/blog/v2/stats",
        destPath: "/v2/stats"
      },
      {
        srcPath: "/_api/v3/posts",
        destPath: "/v3/posts"
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_posts";
function queryPostCountStats(payload) {
  function __queryPostCountStats({ host }) {
    const serializedData = (0, import_transform_paths.transformPaths)(payload, [
      {
        transformFn: import_timestamp.transformSDKTimestampToRESTTimestamp,
        paths: [{ path: "rangeStart" }, { path: "rangeEnd" }]
      }
    ]);
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.BlogStatsService.QueryPostCountStats",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogBlogStatsServiceUrl({
        protoPath: "/v2/stats/post/count",
        data: serializedData,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(serializedData),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "stats.periodStart" }]
        }
      ])
    };
    return metadata;
  }
  return __queryPostCountStats;
}
function getTotalPosts(payload) {
  function __getTotalPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.BlogStatsService.GetTotalPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogBlogStatsServiceUrl({
        protoPath: "/v2/stats/posts/total",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload)
    };
    return metadata;
  }
  return __getTotalPosts;
}
function getPost(payload) {
  function __getPost({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.PostService.GetPost",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3PostServiceUrl({
        protoPath: "/v3/posts/{postId}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "post.firstPublishedDate" },
            { path: "post.lastPublishedDate" },
            { path: "post.coverMedia.image.urlExpirationDate" },
            { path: "post.coverMedia.video.thumbnail.urlExpirationDate" },
            { path: "post.heroImage.urlExpirationDate" },
            { path: "post.richContent.metadata.createdTimestamp" },
            { path: "post.richContent.metadata.updatedTimestamp" },
            { path: "post.moderationDetails.submittedDate" },
            { path: "post.moderationDetails.moderationDate" },
            { path: "post.media.wixMedia.image.urlExpirationDate" },
            { path: "post.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "post.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "post.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            { path: "post.media.wixMedia.videoV2.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "post.richContentCompressed" }]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "post.coverMedia.image.focalPoint.x" },
            { path: "post.coverMedia.image.focalPoint.y" },
            { path: "post.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "post.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "post.heroImage.focalPoint.x" },
            { path: "post.heroImage.focalPoint.y" },
            { path: "post.media.wixMedia.image.focalPoint.x" },
            { path: "post.media.wixMedia.image.focalPoint.y" },
            {
              path: "post.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "post.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "post.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "post.media.wixMedia.videoV2.posters.focalPoint.y" },
            { path: "post.metrics.averageRating" },
            {
              path: "post.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "post.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "post.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            { path: "post.richContent.nodes.galleryData.options.item.ratio" },
            { path: "post.richContent.nodes.imageData.image.duration" },
            { path: "post.richContent.nodes.mapData.mapSettings.lat" },
            { path: "post.richContent.nodes.mapData.mapSettings.lng" },
            { path: "post.richContent.nodes.pollData.poll.image.duration" },
            {
              path: "post.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "post.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "post.richContent.nodes.appEmbedData.image.duration" },
            { path: "post.richContent.nodes.videoData.video.duration" },
            { path: "post.richContent.nodes.videoData.thumbnail.duration" },
            { path: "post.richContent.nodes.audioData.audio.duration" },
            { path: "post.richContent.nodes.audioData.coverImage.duration" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPost;
}
function getPostBySlug(payload) {
  function __getPostBySlug({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.PostService.GetPostBySlug",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3PostServiceUrl({
        protoPath: "/v3/posts/slugs/{slug}",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "post.firstPublishedDate" },
            { path: "post.lastPublishedDate" },
            { path: "post.coverMedia.image.urlExpirationDate" },
            { path: "post.coverMedia.video.thumbnail.urlExpirationDate" },
            { path: "post.heroImage.urlExpirationDate" },
            { path: "post.richContent.metadata.createdTimestamp" },
            { path: "post.richContent.metadata.updatedTimestamp" },
            { path: "post.moderationDetails.submittedDate" },
            { path: "post.moderationDetails.moderationDate" },
            { path: "post.media.wixMedia.image.urlExpirationDate" },
            { path: "post.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "post.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "post.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            { path: "post.media.wixMedia.videoV2.posters.urlExpirationDate" }
          ]
        },
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "post.richContentCompressed" }]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "post.coverMedia.image.focalPoint.x" },
            { path: "post.coverMedia.image.focalPoint.y" },
            { path: "post.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "post.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "post.heroImage.focalPoint.x" },
            { path: "post.heroImage.focalPoint.y" },
            { path: "post.media.wixMedia.image.focalPoint.x" },
            { path: "post.media.wixMedia.image.focalPoint.y" },
            {
              path: "post.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "post.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "post.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "post.media.wixMedia.videoV2.posters.focalPoint.y" },
            { path: "post.metrics.averageRating" },
            {
              path: "post.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "post.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "post.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            { path: "post.richContent.nodes.galleryData.options.item.ratio" },
            { path: "post.richContent.nodes.imageData.image.duration" },
            { path: "post.richContent.nodes.mapData.mapSettings.lat" },
            { path: "post.richContent.nodes.mapData.mapSettings.lng" },
            { path: "post.richContent.nodes.pollData.poll.image.duration" },
            {
              path: "post.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "post.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "post.richContent.nodes.appEmbedData.image.duration" },
            { path: "post.richContent.nodes.videoData.video.duration" },
            { path: "post.richContent.nodes.videoData.thumbnail.duration" },
            { path: "post.richContent.nodes.audioData.audio.duration" },
            { path: "post.richContent.nodes.audioData.coverImage.duration" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __getPostBySlug;
}
function listPosts(payload) {
  function __listPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.PostService.ListPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3PostServiceUrl({
        protoPath: "/v3/posts",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "posts.firstPublishedDate" },
            { path: "posts.lastPublishedDate" },
            { path: "posts.coverMedia.image.urlExpirationDate" },
            { path: "posts.coverMedia.video.thumbnail.urlExpirationDate" },
            { path: "posts.heroImage.urlExpirationDate" },
            { path: "posts.richContent.metadata.createdTimestamp" },
            { path: "posts.richContent.metadata.updatedTimestamp" },
            { path: "posts.moderationDetails.submittedDate" },
            { path: "posts.moderationDetails.moderationDate" },
            { path: "posts.media.wixMedia.image.urlExpirationDate" },
            { path: "posts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "posts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "posts.richContentCompressed" }]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "posts.coverMedia.image.focalPoint.x" },
            { path: "posts.coverMedia.image.focalPoint.y" },
            { path: "posts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "posts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "posts.heroImage.focalPoint.x" },
            { path: "posts.heroImage.focalPoint.y" },
            { path: "posts.media.wixMedia.image.focalPoint.x" },
            { path: "posts.media.wixMedia.image.focalPoint.y" },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "posts.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "posts.media.wixMedia.videoV2.posters.focalPoint.y" },
            { path: "posts.metrics.averageRating" },
            {
              path: "posts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "posts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "posts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "posts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "posts.richContent.nodes.imageData.image.duration" },
            { path: "posts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "posts.richContent.nodes.mapData.mapSettings.lng" },
            { path: "posts.richContent.nodes.pollData.poll.image.duration" },
            {
              path: "posts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "posts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "posts.richContent.nodes.appEmbedData.image.duration" },
            { path: "posts.richContent.nodes.videoData.video.duration" },
            { path: "posts.richContent.nodes.videoData.thumbnail.duration" },
            { path: "posts.richContent.nodes.audioData.audio.duration" },
            { path: "posts.richContent.nodes.audioData.coverImage.duration" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __listPosts;
}
function queryPosts(payload) {
  function __queryPosts({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "POST",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.PostService.QueryPosts",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3PostServiceUrl({
        protoPath: "/v3/posts/query",
        data: payload,
        host
      }),
      data: payload,
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp2.transformRESTTimestampToSDKTimestamp,
          paths: [
            { path: "posts.firstPublishedDate" },
            { path: "posts.lastPublishedDate" },
            { path: "posts.coverMedia.image.urlExpirationDate" },
            { path: "posts.coverMedia.video.thumbnail.urlExpirationDate" },
            { path: "posts.heroImage.urlExpirationDate" },
            { path: "posts.richContent.metadata.createdTimestamp" },
            { path: "posts.richContent.metadata.updatedTimestamp" },
            { path: "posts.moderationDetails.submittedDate" },
            { path: "posts.moderationDetails.moderationDate" },
            { path: "posts.media.wixMedia.image.urlExpirationDate" },
            { path: "posts.media.wixMedia.videoV2.urlExpirationDate" },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.urlExpirationDate"
            },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.poster.urlExpirationDate"
            },
            {
              path: "posts.media.wixMedia.videoV2.posters.urlExpirationDate"
            }
          ]
        },
        {
          transformFn: import_bytes.transformRESTBytesToSDKBytes,
          paths: [{ path: "posts.richContentCompressed" }]
        },
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [
            { path: "posts.coverMedia.image.focalPoint.x" },
            { path: "posts.coverMedia.image.focalPoint.y" },
            { path: "posts.coverMedia.video.thumbnail.focalPoint.x" },
            { path: "posts.coverMedia.video.thumbnail.focalPoint.y" },
            { path: "posts.heroImage.focalPoint.x" },
            { path: "posts.heroImage.focalPoint.y" },
            { path: "posts.media.wixMedia.image.focalPoint.x" },
            { path: "posts.media.wixMedia.image.focalPoint.y" },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.poster.focalPoint.x"
            },
            {
              path: "posts.media.wixMedia.videoV2.resolutions.poster.focalPoint.y"
            },
            { path: "posts.media.wixMedia.videoV2.posters.focalPoint.x" },
            { path: "posts.media.wixMedia.videoV2.posters.focalPoint.y" },
            { path: "posts.metrics.averageRating" },
            {
              path: "posts.richContent.nodes.galleryData.items.image.media.duration"
            },
            {
              path: "posts.richContent.nodes.galleryData.items.video.media.duration"
            },
            {
              path: "posts.richContent.nodes.galleryData.items.video.thumbnail.duration"
            },
            {
              path: "posts.richContent.nodes.galleryData.options.item.ratio"
            },
            { path: "posts.richContent.nodes.imageData.image.duration" },
            { path: "posts.richContent.nodes.mapData.mapSettings.lat" },
            { path: "posts.richContent.nodes.mapData.mapSettings.lng" },
            { path: "posts.richContent.nodes.pollData.poll.image.duration" },
            {
              path: "posts.richContent.nodes.pollData.poll.options.image.duration"
            },
            {
              path: "posts.richContent.nodes.pollData.design.poll.background.image.duration"
            },
            { path: "posts.richContent.nodes.appEmbedData.image.duration" },
            { path: "posts.richContent.nodes.videoData.video.duration" },
            { path: "posts.richContent.nodes.videoData.thumbnail.duration" },
            { path: "posts.richContent.nodes.audioData.audio.duration" },
            { path: "posts.richContent.nodes.audioData.coverImage.duration" }
          ]
        }
      ])
    };
    return metadata;
  }
  return __queryPosts;
}
function getPostMetrics(payload) {
  function __getPostMetrics({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v3.post",
      method: "GET",
      methodFqn: "com.wixpress.npm.communities.platformized.blog.v3.PostService.GetPostMetrics",
      packageName: PACKAGE_NAME,
      url: resolveComWixpressNpmCommunitiesPlatformizedBlogV3PostServiceUrl({
        protoPath: "/v3/posts/{postId}/metrics",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_float.transformRESTFloatToSDKFloat,
          paths: [{ path: "metrics.averageRating" }]
        }
      ])
    };
    return metadata;
  }
  return __getPostMetrics;
}

// src/blog-v3-post-posts.meta.ts
function queryPostCountStats2() {
  const payload = {};
  const getRequestOptions = queryPostCountStats(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/stats/post/count",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getTotalPosts2() {
  const payload = {};
  const getRequestOptions = getTotalPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v2/stats/posts/total",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPost2() {
  const payload = { postId: ":postId" };
  const getRequestOptions = getPost(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/posts/{postId}",
    pathParams: { postId: "postId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPostBySlug2() {
  const payload = { slug: ":slug" };
  const getRequestOptions = getPostBySlug(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/posts/slugs/{slug}",
    pathParams: { slug: "slug" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function listPosts2() {
  const payload = {};
  const getRequestOptions = listPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/posts",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function queryPosts2() {
  const payload = {};
  const getRequestOptions = queryPosts(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "POST",
    path: "/v3/posts/query",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
function getPostMetrics2() {
  const payload = { postId: ":postId" };
  const getRequestOptions = getPostMetrics(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v3/posts/{postId}/metrics",
    pathParams: { postId: "postId" },
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getPost,
  getPostBySlug,
  getPostMetrics,
  getTotalPosts,
  listPosts,
  queryPostCountStats,
  queryPosts
});
