// @wix/auto_sdk_blog_blog-cache/build/internal/es/index.mjs
// src/blog-v1-blog-cache-blog-cache.universal.ts
import { transformError as sdkTransformError } from "@wix/sdk-runtime/transform-error";
import {
  renameKeysFromSDKRequestToRESTRequest,
  renameKeysFromRESTResponseToSDKResponse
} from "@wix/sdk-runtime/rename-all-nested-keys";

// src/blog-v1-blog-cache-blog-cache.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBlogV1BlogCacheServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/blog-cache-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_blog-cache";
function getBlogCache(payload) {
  function __getBlogCache({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v1.blog_cache",
      method: "GET",
      methodFqn: "wix.blog.v1.BlogCacheService.GetBlogCache",
      packageName: PACKAGE_NAME,
      url: resolveWixBlogV1BlogCacheServiceUrl({
        protoPath: "/v1/blog-cache",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "blogCache.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getBlogCache;
}

// src/blog-v1-blog-cache-blog-cache.universal.ts
var LocaleStatus = /* @__PURE__ */ ((LocaleStatus2) => {
  LocaleStatus2["UNDEFINED_STATUS"] = "UNDEFINED_STATUS";
  LocaleStatus2["INACTIVE"] = "INACTIVE";
  LocaleStatus2["ACTIVE"] = "ACTIVE";
  LocaleStatus2["DEFAULT"] = "DEFAULT";
  return LocaleStatus2;
})(LocaleStatus || {});
var Flag = /* @__PURE__ */ ((Flag2) => {
  Flag2["UNDEFINED_FLAG_CODE"] = "UNDEFINED_FLAG_CODE";
  Flag2["AFG"] = "AFG";
  Flag2["AGO"] = "AGO";
  Flag2["AIA"] = "AIA";
  Flag2["ALA"] = "ALA";
  Flag2["ALB"] = "ALB";
  Flag2["AND"] = "AND";
  Flag2["ARE"] = "ARE";
  Flag2["ARG"] = "ARG";
  Flag2["ARM"] = "ARM";
  Flag2["ASM"] = "ASM";
  Flag2["ATF"] = "ATF";
  Flag2["ATG"] = "ATG";
  Flag2["AUS"] = "AUS";
  Flag2["AUT"] = "AUT";
  Flag2["AZE"] = "AZE";
  Flag2["BDI"] = "BDI";
  Flag2["BEL"] = "BEL";
  Flag2["BEN"] = "BEN";
  Flag2["BES"] = "BES";
  Flag2["BFA"] = "BFA";
  Flag2["BGD"] = "BGD";
  Flag2["BGR"] = "BGR";
  Flag2["BHR"] = "BHR";
  Flag2["BHS"] = "BHS";
  Flag2["BIH"] = "BIH";
  Flag2["BLM"] = "BLM";
  Flag2["BLR"] = "BLR";
  Flag2["BLZ"] = "BLZ";
  Flag2["BMU"] = "BMU";
  Flag2["BOL"] = "BOL";
  Flag2["BRA"] = "BRA";
  Flag2["BRB"] = "BRB";
  Flag2["BRN"] = "BRN";
  Flag2["BRT"] = "BRT";
  Flag2["BTN"] = "BTN";
  Flag2["BWA"] = "BWA";
  Flag2["CAF"] = "CAF";
  Flag2["CAN"] = "CAN";
  Flag2["CAT"] = "CAT";
  Flag2["CHE"] = "CHE";
  Flag2["CHL"] = "CHL";
  Flag2["CHN"] = "CHN";
  Flag2["CIV"] = "CIV";
  Flag2["CMR"] = "CMR";
  Flag2["COD"] = "COD";
  Flag2["COG"] = "COG";
  Flag2["COL"] = "COL";
  Flag2["COM"] = "COM";
  Flag2["COR"] = "COR";
  Flag2["CPV"] = "CPV";
  Flag2["CRI"] = "CRI";
  Flag2["CUB"] = "CUB";
  Flag2["CUW"] = "CUW";
  Flag2["CYM"] = "CYM";
  Flag2["CZE"] = "CZE";
  Flag2["DEU"] = "DEU";
  Flag2["DJI"] = "DJI";
  Flag2["DMA"] = "DMA";
  Flag2["DNK"] = "DNK";
  Flag2["DOM"] = "DOM";
  Flag2["DZA"] = "DZA";
  Flag2["ECU"] = "ECU";
  Flag2["EGY"] = "EGY";
  Flag2["ERI"] = "ERI";
  Flag2["ESP"] = "ESP";
  Flag2["EST"] = "EST";
  Flag2["ETH"] = "ETH";
  Flag2["EUR"] = "EUR";
  Flag2["EUS"] = "EUS";
  Flag2["FIN"] = "FIN";
  Flag2["FJI"] = "FJI";
  Flag2["FLK"] = "FLK";
  Flag2["FRA"] = "FRA";
  Flag2["FRO"] = "FRO";
  Flag2["FSM"] = "FSM";
  Flag2["GAB"] = "GAB";
  Flag2["GBR"] = "GBR";
  Flag2["GEO"] = "GEO";
  Flag2["GGY"] = "GGY";
  Flag2["GHA"] = "GHA";
  Flag2["GIB"] = "GIB";
  Flag2["GIN"] = "GIN";
  Flag2["GLG"] = "GLG";
  Flag2["GLO"] = "GLO";
  Flag2["GLP"] = "GLP";
  Flag2["GMB"] = "GMB";
  Flag2["GNB"] = "GNB";
  Flag2["GNQ"] = "GNQ";
  Flag2["GRC"] = "GRC";
  Flag2["GRD"] = "GRD";
  Flag2["GRL"] = "GRL";
  Flag2["GTM"] = "GTM";
  Flag2["GUM"] = "GUM";
  Flag2["GUY"] = "GUY";
  Flag2["HKG"] = "HKG";
  Flag2["HND"] = "HND";
  Flag2["HRV"] = "HRV";
  Flag2["HTI"] = "HTI";
  Flag2["HUN"] = "HUN";
  Flag2["IDN"] = "IDN";
  Flag2["IMN"] = "IMN";
  Flag2["IND"] = "IND";
  Flag2["IOT"] = "IOT";
  Flag2["IRL"] = "IRL";
  Flag2["IRN"] = "IRN";
  Flag2["IRQ"] = "IRQ";
  Flag2["ISL"] = "ISL";
  Flag2["ISR"] = "ISR";
  Flag2["ITA"] = "ITA";
  Flag2["JAM"] = "JAM";
  Flag2["JEY"] = "JEY";
  Flag2["JOR"] = "JOR";
  Flag2["JPN"] = "JPN";
  Flag2["KAZ"] = "KAZ";
  Flag2["KEN"] = "KEN";
  Flag2["KGZ"] = "KGZ";
  Flag2["KHM"] = "KHM";
  Flag2["KIR"] = "KIR";
  Flag2["KNA"] = "KNA";
  Flag2["KOR"] = "KOR";
  Flag2["KUR"] = "KUR";
  Flag2["KWT"] = "KWT";
  Flag2["LAO"] = "LAO";
  Flag2["LBN"] = "LBN";
  Flag2["LBR"] = "LBR";
  Flag2["LBY"] = "LBY";
  Flag2["LCA"] = "LCA";
  Flag2["LIE"] = "LIE";
  Flag2["LKA"] = "LKA";
  Flag2["LSO"] = "LSO";
  Flag2["LTU"] = "LTU";
  Flag2["LUX"] = "LUX";
  Flag2["LVA"] = "LVA";
  Flag2["MAC"] = "MAC";
  Flag2["MAF"] = "MAF";
  Flag2["MAR"] = "MAR";
  Flag2["MCO"] = "MCO";
  Flag2["MDA"] = "MDA";
  Flag2["MDG"] = "MDG";
  Flag2["MDV"] = "MDV";
  Flag2["MEX"] = "MEX";
  Flag2["MHL"] = "MHL";
  Flag2["MKD"] = "MKD";
  Flag2["MLI"] = "MLI";
  Flag2["MLT"] = "MLT";
  Flag2["MMR"] = "MMR";
  Flag2["MNE"] = "MNE";
  Flag2["MNG"] = "MNG";
  Flag2["MNP"] = "MNP";
  Flag2["MOZ"] = "MOZ";
  Flag2["MRT"] = "MRT";
  Flag2["MSR"] = "MSR";
  Flag2["MTQ"] = "MTQ";
  Flag2["MWI"] = "MWI";
  Flag2["MYS"] = "MYS";
  Flag2["MYT"] = "MYT";
  Flag2["NAM"] = "NAM";
  Flag2["NCL"] = "NCL";
  Flag2["NER"] = "NER";
  Flag2["NFK"] = "NFK";
  Flag2["NGA"] = "NGA";
  Flag2["NIC"] = "NIC";
  Flag2["NIU"] = "NIU";
  Flag2["NLD"] = "NLD";
  Flag2["NOR"] = "NOR";
  Flag2["NPL"] = "NPL";
  Flag2["NRU"] = "NRU";
  Flag2["NZL"] = "NZL";
  Flag2["OMN"] = "OMN";
  Flag2["PAK"] = "PAK";
  Flag2["PAN"] = "PAN";
  Flag2["PCN"] = "PCN";
  Flag2["PER"] = "PER";
  Flag2["PHL"] = "PHL";
  Flag2["PLW"] = "PLW";
  Flag2["PNG"] = "PNG";
  Flag2["POL"] = "POL";
  Flag2["PRI"] = "PRI";
  Flag2["PRK"] = "PRK";
  Flag2["PRT"] = "PRT";
  Flag2["PRY"] = "PRY";
  Flag2["PSE"] = "PSE";
  Flag2["PYF"] = "PYF";
  Flag2["QAT"] = "QAT";
  Flag2["REU"] = "REU";
  Flag2["ROU"] = "ROU";
  Flag2["RUS"] = "RUS";
  Flag2["RWA"] = "RWA";
  Flag2["SAU"] = "SAU";
  Flag2["SDN"] = "SDN";
  Flag2["SEN"] = "SEN";
  Flag2["SGP"] = "SGP";
  Flag2["SGS"] = "SGS";
  Flag2["SHN"] = "SHN";
  Flag2["SJM"] = "SJM";
  Flag2["SLB"] = "SLB";
  Flag2["SLE"] = "SLE";
  Flag2["SLV"] = "SLV";
  Flag2["SMR"] = "SMR";
  Flag2["SOM"] = "SOM";
  Flag2["SPM"] = "SPM";
  Flag2["SRB"] = "SRB";
  Flag2["SSD"] = "SSD";
  Flag2["STP"] = "STP";
  Flag2["SUR"] = "SUR";
  Flag2["SVK"] = "SVK";
  Flag2["SVN"] = "SVN";
  Flag2["SWE"] = "SWE";
  Flag2["SWZ"] = "SWZ";
  Flag2["SXM"] = "SXM";
  Flag2["SYC"] = "SYC";
  Flag2["SYR"] = "SYR";
  Flag2["TCA"] = "TCA";
  Flag2["TCD"] = "TCD";
  Flag2["TGO"] = "TGO";
  Flag2["THA"] = "THA";
  Flag2["TJK"] = "TJK";
  Flag2["TKL"] = "TKL";
  Flag2["TKM"] = "TKM";
  Flag2["TLS"] = "TLS";
  Flag2["TON"] = "TON";
  Flag2["TTO"] = "TTO";
  Flag2["TUN"] = "TUN";
  Flag2["TUR"] = "TUR";
  Flag2["TUV"] = "TUV";
  Flag2["TWN"] = "TWN";
  Flag2["TZA"] = "TZA";
  Flag2["UGA"] = "UGA";
  Flag2["UKR"] = "UKR";
  Flag2["URY"] = "URY";
  Flag2["USA"] = "USA";
  Flag2["UZB"] = "UZB";
  Flag2["VAT"] = "VAT";
  Flag2["VCT"] = "VCT";
  Flag2["VEN"] = "VEN";
  Flag2["VGB"] = "VGB";
  Flag2["VIR"] = "VIR";
  Flag2["VLC"] = "VLC";
  Flag2["VNM"] = "VNM";
  Flag2["VUT"] = "VUT";
  Flag2["WLF"] = "WLF";
  Flag2["WLS"] = "WLS";
  Flag2["YEM"] = "YEM";
  Flag2["ZAF"] = "ZAF";
  Flag2["ZMB"] = "ZMB";
  Flag2["ZWE"] = "ZWE";
  return Flag2;
})(Flag || {});
var ResolutionMethod = /* @__PURE__ */ ((ResolutionMethod2) => {
  ResolutionMethod2["QUERY_PARAM"] = "QUERY_PARAM";
  ResolutionMethod2["SUBDOMAIN"] = "SUBDOMAIN";
  ResolutionMethod2["SUBDIRECTORY"] = "SUBDIRECTORY";
  return ResolutionMethod2;
})(ResolutionMethod || {});
var PlacementType = /* @__PURE__ */ ((PlacementType2) => {
  PlacementType2["BEFORE"] = "BEFORE";
  PlacementType2["AFTER"] = "AFTER";
  PlacementType2["REPLACE"] = "REPLACE";
  return PlacementType2;
})(PlacementType || {});
var DayOfWeek = /* @__PURE__ */ ((DayOfWeek2) => {
  DayOfWeek2["MONDAY"] = "MONDAY";
  DayOfWeek2["TUESDAY"] = "TUESDAY";
  DayOfWeek2["WEDNESDAY"] = "WEDNESDAY";
  DayOfWeek2["THURSDAY"] = "THURSDAY";
  DayOfWeek2["FRIDAY"] = "FRIDAY";
  DayOfWeek2["SATURDAY"] = "SATURDAY";
  DayOfWeek2["SUNDAY"] = "SUNDAY";
  return DayOfWeek2;
})(DayOfWeek || {});
async function getBlogCache2() {
  const { httpClient, sideEffects } = arguments[0];
  const payload = renameKeysFromSDKRequestToRESTRequest({});
  const reqOpts = getBlogCache(payload);
  sideEffects?.onSiteCall?.();
  try {
    const result = await httpClient.request(reqOpts);
    sideEffects?.onSuccess?.(result);
    return renameKeysFromRESTResponseToSDKResponse(result.data);
  } catch (err) {
    const transformedError = sdkTransformError(
      err,
      {
        spreadPathsToArguments: {},
        explicitPathsToArguments: {},
        singleArgumentUnchanged: false
      },
      []
    );
    sideEffects?.onError?.(err);
    throw transformedError;
  }
}

// src/blog-v1-blog-cache-blog-cache.public.ts
function getBlogCache3(httpClient) {
  return () => getBlogCache2(
    // @ts-ignore
    { httpClient }
  );
}

// src/blog-v1-blog-cache-blog-cache.context.ts
import { createRESTModule } from "@wix/sdk-runtime/rest-modules";
var getBlogCache4 = /* @__PURE__ */ createRESTModule(getBlogCache3);
export {
  DayOfWeek,
  Flag,
  LocaleStatus,
  PlacementType,
  ResolutionMethod,
  getBlogCache4 as getBlogCache
};
