// @wix/auto_sdk_blog_blog-cache/build/internal/es/meta.mjs
// src/blog-v1-blog-cache-blog-cache.http.ts
import { toURLSearchParams } from "@wix/sdk-runtime/rest-modules";
import { transformRESTTimestampToSDKTimestamp } from "@wix/sdk-runtime/transformations/timestamp";
import { transformPaths } from "@wix/sdk-runtime/transformations/transform-paths";
import { resolveUrl } from "@wix/sdk-runtime/rest-modules";
function resolveWixBlogV1BlogCacheServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/blog-cache-service",
        destPath: ""
      }
    ]
  };
  return resolveUrl(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_blog-cache";
function getBlogCache(payload) {
  function __getBlogCache({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v1.blog_cache",
      method: "GET",
      methodFqn: "wix.blog.v1.BlogCacheService.GetBlogCache",
      packageName: PACKAGE_NAME,
      url: resolveWixBlogV1BlogCacheServiceUrl({
        protoPath: "/v1/blog-cache",
        data: payload,
        host
      }),
      params: toURLSearchParams(payload),
      transformResponse: (payload2) => transformPaths(payload2, [
        {
          transformFn: transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "blogCache.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getBlogCache;
}

// src/blog-v1-blog-cache-blog-cache.meta.ts
function getBlogCache2() {
  const payload = {};
  const getRequestOptions = getBlogCache(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/blog-cache",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
export {
  getBlogCache2 as getBlogCache
};
