// @wix/auto_sdk_blog_blog-cache/build/internal/cjs/meta.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// meta.ts
var meta_exports = {};
__export(meta_exports, {
  getBlogCache: () => getBlogCache2
});
module.exports = __toCommonJS(meta_exports);

// src/blog-v1-blog-cache-blog-cache.http.ts
var import_rest_modules = require("@wix/sdk-runtime/rest-modules");
var import_timestamp = require("@wix/sdk-runtime/transformations/timestamp");
var import_transform_paths = require("@wix/sdk-runtime/transformations/transform-paths");
var import_rest_modules2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixBlogV1BlogCacheServiceUrl(opts) {
  const domainToMappings = {
    "bo._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "wixbo.ai": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "wix-bo.com": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "manage._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "apps._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "editor._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "blocks._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "create.editorx": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    _: [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "www._base_domain_": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "*.dev.wix-code.com": [
      {
        srcPath: "/_api/blog-cache-service",
        destPath: ""
      }
    ],
    "api._api_base_domain_": [
      {
        srcPath: "/blog-cache-service",
        destPath: ""
      }
    ]
  };
  return (0, import_rest_modules2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
var PACKAGE_NAME = "@wix/auto_sdk_blog_blog-cache";
function getBlogCache(payload) {
  function __getBlogCache({ host }) {
    const metadata = {
      entityFqdn: "wix.blog.v1.blog_cache",
      method: "GET",
      methodFqn: "wix.blog.v1.BlogCacheService.GetBlogCache",
      packageName: PACKAGE_NAME,
      url: resolveWixBlogV1BlogCacheServiceUrl({
        protoPath: "/v1/blog-cache",
        data: payload,
        host
      }),
      params: (0, import_rest_modules.toURLSearchParams)(payload),
      transformResponse: (payload2) => (0, import_transform_paths.transformPaths)(payload2, [
        {
          transformFn: import_timestamp.transformRESTTimestampToSDKTimestamp,
          paths: [{ path: "blogCache.updatedDate" }]
        }
      ])
    };
    return metadata;
  }
  return __getBlogCache;
}

// src/blog-v1-blog-cache-blog-cache.meta.ts
function getBlogCache2() {
  const payload = {};
  const getRequestOptions = getBlogCache(payload);
  const getUrl = (context) => {
    const { url } = getRequestOptions(context);
    return url;
  };
  return {
    getUrl,
    httpMethod: "GET",
    path: "/v1/blog-cache",
    pathParams: {},
    __requestType: null,
    __originalRequestType: null,
    __responseType: null,
    __originalResponseType: null
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getBlogCache
});
